{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "732ba631-15d8-45a0-acae-5f8ba1eb8fe7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e756bb38-4e5b-49b0-8055-2c4ccae6c4c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df = pd.read_csv(\"GSE68086_series_matrix.csv\")\n",
    "tep_df = pd.read_csv(\"GSE68086_TEP_data_matrix.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ea1c1408-e4a5-45c6-b9f0-4cf63b39706f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 285 entries, 0 to 284\n",
      "Data columns (total 46 columns):\n",
      " #   Column                           Non-Null Count  Dtype  \n",
      "---  ------                           --------------  -----  \n",
      " 0   !Sample_geo_accession            285 non-null    object \n",
      " 1   !Sample_status                   285 non-null    object \n",
      " 2   !Sample_submission_date          285 non-null    object \n",
      " 3   !Sample_last_update_date         285 non-null    object \n",
      " 4   !Sample_type                     285 non-null    object \n",
      " 5   !Sample_channel_count            285 non-null    object \n",
      " 6   !Sample_source_name_ch1          285 non-null    object \n",
      " 7   !Sample_organism_ch1             285 non-null    object \n",
      " 8   !Sample_characteristics_ch1      285 non-null    object \n",
      " 9   !Sample_characteristics_ch1.1    285 non-null    object \n",
      " 10  !Sample_characteristics_ch1.2    285 non-null    object \n",
      " 11  !Sample_characteristics_ch1.3    285 non-null    object \n",
      " 12  !Sample_characteristics_ch1.4    285 non-null    object \n",
      " 13  !Sample_characteristics_ch1.5    285 non-null    object \n",
      " 14  !Sample_molecule_ch1             285 non-null    object \n",
      " 15  !Sample_extract_protocol_ch1     285 non-null    object \n",
      " 16  !Sample_extract_protocol_ch1.1   285 non-null    object \n",
      " 17  !Sample_taxid_ch1                285 non-null    object \n",
      " 18  !Sample_description              285 non-null    object \n",
      " 19  !Sample_data_processing          285 non-null    object \n",
      " 20  !Sample_data_processing.1        285 non-null    object \n",
      " 21  !Sample_data_processing.2        285 non-null    object \n",
      " 22  !Sample_data_processing.3        285 non-null    object \n",
      " 23  !Sample_data_processing.4        285 non-null    object \n",
      " 24  !Sample_data_processing.5        285 non-null    object \n",
      " 25  !Sample_platform_id              285 non-null    object \n",
      " 26  !Sample_contact_name             285 non-null    object \n",
      " 27  !Sample_contact_email            285 non-null    object \n",
      " 28  !Sample_contact_laboratory       285 non-null    object \n",
      " 29  !Sample_contact_department       285 non-null    object \n",
      " 30  !Sample_contact_institute        285 non-null    object \n",
      " 31  !Sample_contact_address          285 non-null    object \n",
      " 32  !Sample_contact_city             285 non-null    object \n",
      " 33  !Sample_contact_zip/postal_code  285 non-null    object \n",
      " 34  !Sample_contact_country          285 non-null    object \n",
      " 35  !Sample_data_row_count           285 non-null    object \n",
      " 36  !Sample_instrument_model         285 non-null    object \n",
      " 37  !Sample_library_selection        285 non-null    object \n",
      " 38  !Sample_library_source           285 non-null    object \n",
      " 39  !Sample_library_strategy         285 non-null    object \n",
      " 40  !Sample_relation                 285 non-null    object \n",
      " 41  !Sample_relation.1               285 non-null    object \n",
      " 42  !Sample_supplementary_file_1     285 non-null    object \n",
      " 43  !series_matrix_table_begin       0 non-null      float64\n",
      " 44  \"ID_REF\"                         285 non-null    object \n",
      " 45  !series_matrix_table_end         0 non-null      float64\n",
      "dtypes: float64(2), object(44)\n",
      "memory usage: 102.6+ KB\n"
     ]
    }
   ],
   "source": [
    "series_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6dedd8fd-50ca-4eee-97b3-1246bb5c3364",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>3-Breast-Her2-ampl</th>\n",
       "      <th>8-Breast-WT</th>\n",
       "      <th>10-Breast-Her2-ampl</th>\n",
       "      <th>Breast-100</th>\n",
       "      <th>15-Breast-Her2-ampl</th>\n",
       "      <th>16-Breast-WT</th>\n",
       "      <th>21-Breast-WT</th>\n",
       "      <th>33-Breast-Her2-ampl</th>\n",
       "      <th>42-Breast-Her2-ampl</th>\n",
       "      <th>Breast-454</th>\n",
       "      <th>48-Breast-WT</th>\n",
       "      <th>50-Breast-WT</th>\n",
       "      <th>55-Breast-WT</th>\n",
       "      <th>Type-Unknown-6</th>\n",
       "      <th>Breast-564</th>\n",
       "      <th>59-Breast-WT</th>\n",
       "      <th>Breast-590</th>\n",
       "      <th>Breast-60</th>\n",
       "      <th>Breast-603</th>\n",
       "      <th>ALK61-Breast-Her2-unknown</th>\n",
       "      <th>Breast-85</th>\n",
       "      <th>Breast-86</th>\n",
       "      <th>97-Breast-Her2-ampl</th>\n",
       "      <th>Breast-ALK-82</th>\n",
       "      <th>292-Liver-KRAS</th>\n",
       "      <th>316-Liver-KRAS</th>\n",
       "      <th>341-Liver-WT</th>\n",
       "      <th>376-Liver-WT</th>\n",
       "      <th>379-Liver-WT</th>\n",
       "      <th>Chol-410</th>\n",
       "      <th>Chol-442</th>\n",
       "      <th>Chol-460</th>\n",
       "      <th>Chol-611</th>\n",
       "      <th>Chol-ALK-28</th>\n",
       "      <th>VU256-CRC</th>\n",
       "      <th>VU258-CRC</th>\n",
       "      <th>VU272-CRC</th>\n",
       "      <th>VU276-CRC</th>\n",
       "      <th>Type-Unknown-1</th>\n",
       "      <th>329-CRC-KRAS</th>\n",
       "      <th>342-CRC-KRAS</th>\n",
       "      <th>Type-Unknown-5</th>\n",
       "      <th>354-CRC-WT</th>\n",
       "      <th>356-CRC-WT</th>\n",
       "      <th>CRC-357</th>\n",
       "      <th>374-CRC-WT</th>\n",
       "      <th>378-CRC-WT</th>\n",
       "      <th>420-CRC-WT</th>\n",
       "      <th>429-CRC-WT</th>\n",
       "      <th>CRC-450</th>\n",
       "      <th>453-CRC-KRAS</th>\n",
       "      <th>CRC-459</th>\n",
       "      <th>462-CRC-KRAS</th>\n",
       "      <th>CRC-463</th>\n",
       "      <th>466-CRC-KRAS</th>\n",
       "      <th>474-CRC-WT</th>\n",
       "      <th>CRC-487</th>\n",
       "      <th>CRC-496</th>\n",
       "      <th>497-CRC-KRAS</th>\n",
       "      <th>504-CRC-WT</th>\n",
       "      <th>CRC-511</th>\n",
       "      <th>CRC-523</th>\n",
       "      <th>CRC-524</th>\n",
       "      <th>527-CRC-WT</th>\n",
       "      <th>CRC-531</th>\n",
       "      <th>543-CRC-KRAS</th>\n",
       "      <th>546-CRC-KRAS</th>\n",
       "      <th>548-CRC-KRAS</th>\n",
       "      <th>CRC-552</th>\n",
       "      <th>556-CRC-WT</th>\n",
       "      <th>CRC-565</th>\n",
       "      <th>CRC-576</th>\n",
       "      <th>VU280-GBM</th>\n",
       "      <th>VU284-GBM-vIII</th>\n",
       "      <th>VU306-GBM-vIII</th>\n",
       "      <th>VU360-GBM-WT</th>\n",
       "      <th>VU372-GBM-vIII</th>\n",
       "      <th>VU378-GBM</th>\n",
       "      <th>VU383Platelet-hiseq</th>\n",
       "      <th>VU393-GBM-vIII</th>\n",
       "      <th>VU394Platelet-hiseq</th>\n",
       "      <th>VU398-GBM</th>\n",
       "      <th>VU402-2-GBM</th>\n",
       "      <th>VU406-2-GBM</th>\n",
       "      <th>VU408-GBM</th>\n",
       "      <th>VU417-GBM-vIII</th>\n",
       "      <th>VU418-GBM</th>\n",
       "      <th>VU419-GBM</th>\n",
       "      <th>VU421-GBM</th>\n",
       "      <th>VU422-GBM</th>\n",
       "      <th>VU425-GBM</th>\n",
       "      <th>VU429-1-GBM-vIII</th>\n",
       "      <th>VU430-GBM</th>\n",
       "      <th>VU431-1-GBM-vIII</th>\n",
       "      <th>VU436-1-GBM-vIII</th>\n",
       "      <th>GBM-438</th>\n",
       "      <th>VU439-1-GBM-vIII</th>\n",
       "      <th>GBM-440</th>\n",
       "      <th>VU443-1-GBM-WT</th>\n",
       "      <th>VU445-1-GBM-WT</th>\n",
       "      <th>VU454-1-GBM-WT</th>\n",
       "      <th>VU456-1-GBM-WT</th>\n",
       "      <th>VU468-1-GBM-WT</th>\n",
       "      <th>GBM-471</th>\n",
       "      <th>GBM-475</th>\n",
       "      <th>GBM-476</th>\n",
       "      <th>GBM-480</th>\n",
       "      <th>GBM-484</th>\n",
       "      <th>GBM-485</th>\n",
       "      <th>GBM-486</th>\n",
       "      <th>GBM-499</th>\n",
       "      <th>GBM-505</th>\n",
       "      <th>HD-1</th>\n",
       "      <th>HD-2-1</th>\n",
       "      <th>HD-3-1</th>\n",
       "      <th>HD-4</th>\n",
       "      <th>HD-5</th>\n",
       "      <th>HD-7</th>\n",
       "      <th>HD-8</th>\n",
       "      <th>HD-9</th>\n",
       "      <th>HD-10</th>\n",
       "      <th>HD-11-1</th>\n",
       "      <th>HD-13-1</th>\n",
       "      <th>HD-14-1</th>\n",
       "      <th>HD-15-1</th>\n",
       "      <th>HD-16-1</th>\n",
       "      <th>HD-20-1</th>\n",
       "      <th>HD-22-1</th>\n",
       "      <th>HD-34-1</th>\n",
       "      <th>HD-39-1</th>\n",
       "      <th>HD-40-1</th>\n",
       "      <th>HD-45-1</th>\n",
       "      <th>HD-46-1</th>\n",
       "      <th>Control-2</th>\n",
       "      <th>Control-3</th>\n",
       "      <th>Control-4</th>\n",
       "      <th>Control-5</th>\n",
       "      <th>Control-6</th>\n",
       "      <th>Control-8</th>\n",
       "      <th>Control-9</th>\n",
       "      <th>Control-10</th>\n",
       "      <th>Control-13</th>\n",
       "      <th>Type-Unknown-3</th>\n",
       "      <th>HD-17-1</th>\n",
       "      <th>HD-18-2</th>\n",
       "      <th>HD-21-1</th>\n",
       "      <th>HD-24-2</th>\n",
       "      <th>HD-25-2</th>\n",
       "      <th>HD-26-2</th>\n",
       "      <th>HD-27-2</th>\n",
       "      <th>HD-29-2</th>\n",
       "      <th>HD-30-2</th>\n",
       "      <th>HD-31</th>\n",
       "      <th>HD-32-2</th>\n",
       "      <th>HD-33-1</th>\n",
       "      <th>HD-35-2</th>\n",
       "      <th>HD-36</th>\n",
       "      <th>HD-37-2</th>\n",
       "      <th>HD-38-2</th>\n",
       "      <th>HD-44-2</th>\n",
       "      <th>HD-47-1</th>\n",
       "      <th>HD-48-1</th>\n",
       "      <th>HD-49-1</th>\n",
       "      <th>HD-50-1</th>\n",
       "      <th>HD-51-1</th>\n",
       "      <th>HD-53-1</th>\n",
       "      <th>HD-54-1</th>\n",
       "      <th>VU274-CRC</th>\n",
       "      <th>297-Liver-WT</th>\n",
       "      <th>366-Liver-WT</th>\n",
       "      <th>Lung-0082</th>\n",
       "      <th>Lung-0096</th>\n",
       "      <th>Lung-0001</th>\n",
       "      <th>Lung-0100</th>\n",
       "      <th>Lung-0101</th>\n",
       "      <th>Lung-0102</th>\n",
       "      <th>Lung-0104</th>\n",
       "      <th>Lung-0110</th>\n",
       "      <th>Lung-0112</th>\n",
       "      <th>Lung-0119</th>\n",
       "      <th>Lung-004-EGFR</th>\n",
       "      <th>Lung-006</th>\n",
       "      <th>Lung-0007</th>\n",
       "      <th>Lung-008</th>\n",
       "      <th>Lung-0012</th>\n",
       "      <th>Lung-0014J</th>\n",
       "      <th>Lung-0015</th>\n",
       "      <th>Lung-0016</th>\n",
       "      <th>Lung-0017</th>\n",
       "      <th>Lung-0020J</th>\n",
       "      <th>Lung-0020J-1</th>\n",
       "      <th>Lung-0024J-1</th>\n",
       "      <th>Lung-0025J</th>\n",
       "      <th>Lung-0026J</th>\n",
       "      <th>VU271-CRC</th>\n",
       "      <th>Lung-0029J-1-WT</th>\n",
       "      <th>Lung-0030J-1-WT</th>\n",
       "      <th>Lung-0037</th>\n",
       "      <th>Lung-0040</th>\n",
       "      <th>Lung-0042</th>\n",
       "      <th>Lung-0046</th>\n",
       "      <th>Lung-0058-EGFR</th>\n",
       "      <th>Lung-0061-EGFR</th>\n",
       "      <th>Lung-0062-EGFR</th>\n",
       "      <th>Lung-0074-EGFR</th>\n",
       "      <th>Lung-L-13</th>\n",
       "      <th>Lung-L-15</th>\n",
       "      <th>Lung-L-26</th>\n",
       "      <th>Lung-L-33</th>\n",
       "      <th>Lung-L-4</th>\n",
       "      <th>VU260-CRC</th>\n",
       "      <th>286-Pancr-286-WT</th>\n",
       "      <th>314-Pancr-WT</th>\n",
       "      <th>327-Pancr-KRAS</th>\n",
       "      <th>328-Pancr-WT</th>\n",
       "      <th>331-Pancr-WT</th>\n",
       "      <th>Panc-335</th>\n",
       "      <th>367-Pancr-KRAS</th>\n",
       "      <th>369-Pancr-KRAS</th>\n",
       "      <th>Panc-370</th>\n",
       "      <th>371-Pancr-KRAS</th>\n",
       "      <th>380-Pancr-KRAS</th>\n",
       "      <th>382-Pancr-KRAS</th>\n",
       "      <th>387-Pancr-WT</th>\n",
       "      <th>388-Pancr-WT</th>\n",
       "      <th>389-Pancr-WT</th>\n",
       "      <th>390-Pancr-KRAS</th>\n",
       "      <th>391-Pancr-KRAS</th>\n",
       "      <th>431-Pancr-WT</th>\n",
       "      <th>434-Pancr-WT</th>\n",
       "      <th>Panc-435</th>\n",
       "      <th>448-Pancr-KRAS</th>\n",
       "      <th>Type-Unknown-4</th>\n",
       "      <th>Panc-451</th>\n",
       "      <th>Panc-461</th>\n",
       "      <th>Panc-477</th>\n",
       "      <th>488-Pancr-WT</th>\n",
       "      <th>Panc-493</th>\n",
       "      <th>498-Pancr-WT</th>\n",
       "      <th>Panc-501</th>\n",
       "      <th>Panc-503</th>\n",
       "      <th>512-Pancr-KRAS</th>\n",
       "      <th>553-Pancr-KRAS</th>\n",
       "      <th>Panc-593</th>\n",
       "      <th>Panc-597</th>\n",
       "      <th>MGH-BrCa-H-11</th>\n",
       "      <th>MGH-BrCa-H-59</th>\n",
       "      <th>MGH-BrCa-H-66</th>\n",
       "      <th>MGH-BrCa-H-68</th>\n",
       "      <th>MGH-BrCa-H-74</th>\n",
       "      <th>MGH-BrCa-H-75</th>\n",
       "      <th>MGH-CRC-368</th>\n",
       "      <th>Vumc-NSCLC-49</th>\n",
       "      <th>Vumc-NSCLC-57</th>\n",
       "      <th>Vumc-NSCLC-59</th>\n",
       "      <th>MGH-BrCa-H31-TR-489</th>\n",
       "      <th>MGH-BrCa-H38-TR467</th>\n",
       "      <th>MGH-BrCa-H63-TR-491</th>\n",
       "      <th>MGH-BrCa-H76-TR469</th>\n",
       "      <th>MGH-BrCa-H81-TR492</th>\n",
       "      <th>MGH-BrCa-H83-TR550</th>\n",
       "      <th>MGH-BrCa-H84-TR551</th>\n",
       "      <th>MGH-BrCa-H89-TR471</th>\n",
       "      <th>MGH-BrCa-H92-TR472</th>\n",
       "      <th>MGH-CRC-412-TR466</th>\n",
       "      <th>MGH-CRC-BRAF4-TR547</th>\n",
       "      <th>MGH-CRC-BRAF5-TR548</th>\n",
       "      <th>MGH-CRC-BRAF6-TR465</th>\n",
       "      <th>MGH-NSCLC-L01-TR458</th>\n",
       "      <th>MGH-NSCLC-L03-TR488</th>\n",
       "      <th>MGH-NSCLC-L07-TR459</th>\n",
       "      <th>MGH-NSCLC-L11-TR477</th>\n",
       "      <th>MGH-NSCLC-L12-TR478</th>\n",
       "      <th>MGH-NSCLC-L18-TR479</th>\n",
       "      <th>MGH-NSCLC-L19-TR461</th>\n",
       "      <th>MGH-NSCLC-L20-TR500</th>\n",
       "      <th>MGH-NSCLC-L22-TR462</th>\n",
       "      <th>MGH-NSCLC-L23-TR524</th>\n",
       "      <th>MGH-NSCLC-L25-TR480</th>\n",
       "      <th>MGH-NSCLC-L39-TR519</th>\n",
       "      <th>MGH-NSCLC-L40-TR520</th>\n",
       "      <th>MGH-NSCLC-L51-TR521</th>\n",
       "      <th>MGH-NSCLC-L58-TR525</th>\n",
       "      <th>MGH-NSCLC-L59-TR522</th>\n",
       "      <th>MGH-NSCLC-L65-TR523</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>241</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>44</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>139</td>\n",
       "      <td>108</td>\n",
       "      <td>55</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>102</td>\n",
       "      <td>93</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>41</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>39</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>61</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>43</td>\n",
       "      <td>5</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>180</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>14</td>\n",
       "      <td>28</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>63</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>41</td>\n",
       "      <td>31</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>84</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "      <td>31</td>\n",
       "      <td>95</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>76</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>105</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>16</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>12</td>\n",
       "      <td>30</td>\n",
       "      <td>54</td>\n",
       "      <td>37</td>\n",
       "      <td>33</td>\n",
       "      <td>56</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>27</td>\n",
       "      <td>22</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>88</td>\n",
       "      <td>6</td>\n",
       "      <td>146</td>\n",
       "      <td>128</td>\n",
       "      <td>32</td>\n",
       "      <td>129</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>131</td>\n",
       "      <td>34</td>\n",
       "      <td>184</td>\n",
       "      <td>214</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>139</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>94</td>\n",
       "      <td>190</td>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>95</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>49</td>\n",
       "      <td>83</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>56</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>156</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "      <td>39</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>196</td>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>247</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>85</td>\n",
       "      <td>57</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>25</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>16</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>39</td>\n",
       "      <td>20</td>\n",
       "      <td>79</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>92</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>40</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>18</td>\n",
       "      <td>40</td>\n",
       "      <td>6</td>\n",
       "      <td>32</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>27</td>\n",
       "      <td>46</td>\n",
       "      <td>42</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>2</td>\n",
       "      <td>41</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>34</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>39</td>\n",
       "      <td>64</td>\n",
       "      <td>7</td>\n",
       "      <td>132</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>97</td>\n",
       "      <td>70</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>90</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>81</td>\n",
       "      <td>98</td>\n",
       "      <td>18</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>144</td>\n",
       "      <td>26</td>\n",
       "      <td>97</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>85</td>\n",
       "      <td>128</td>\n",
       "      <td>28</td>\n",
       "      <td>224</td>\n",
       "      <td>120</td>\n",
       "      <td>21</td>\n",
       "      <td>16</td>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>80</td>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>43</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>22</td>\n",
       "      <td>35</td>\n",
       "      <td>16</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>112</td>\n",
       "      <td>12</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>89</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>93</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>56</td>\n",
       "      <td>21</td>\n",
       "      <td>32</td>\n",
       "      <td>59</td>\n",
       "      <td>43</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>45</td>\n",
       "      <td>47</td>\n",
       "      <td>14</td>\n",
       "      <td>99</td>\n",
       "      <td>20</td>\n",
       "      <td>28</td>\n",
       "      <td>31</td>\n",
       "      <td>34</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>100</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>27</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>29</td>\n",
       "      <td>12</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>60</td>\n",
       "      <td>52</td>\n",
       "      <td>10</td>\n",
       "      <td>95</td>\n",
       "      <td>6</td>\n",
       "      <td>49</td>\n",
       "      <td>8</td>\n",
       "      <td>56</td>\n",
       "      <td>26</td>\n",
       "      <td>168</td>\n",
       "      <td>44</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>25</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>33</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>31</td>\n",
       "      <td>50</td>\n",
       "      <td>20</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>39</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>22</td>\n",
       "      <td>107</td>\n",
       "      <td>21</td>\n",
       "      <td>139</td>\n",
       "      <td>125</td>\n",
       "      <td>43</td>\n",
       "      <td>22</td>\n",
       "      <td>114</td>\n",
       "      <td>31</td>\n",
       "      <td>16</td>\n",
       "      <td>82</td>\n",
       "      <td>40</td>\n",
       "      <td>14</td>\n",
       "      <td>65</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>10</td>\n",
       "      <td>37</td>\n",
       "      <td>34</td>\n",
       "      <td>20</td>\n",
       "      <td>37</td>\n",
       "      <td>34</td>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>8</td>\n",
       "      <td>37</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>38</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>37</td>\n",
       "      <td>41</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>79</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>40</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>50</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0  3-Breast-Her2-ampl  8-Breast-WT  10-Breast-Her2-ampl  \\\n",
       "0  ENSG00000000003                   0            0                    0   \n",
       "1  ENSG00000000005                   0            0                    0   \n",
       "2  ENSG00000000419                  44           14                   16   \n",
       "3  ENSG00000000457                  26            1                   14   \n",
       "4  ENSG00000000460                  81           98                   18   \n",
       "\n",
       "   Breast-100  15-Breast-Her2-ampl  16-Breast-WT  21-Breast-WT  \\\n",
       "0           0                   17             0             0   \n",
       "1           0                    0             0             0   \n",
       "2           8                    9             0           139   \n",
       "3           0                    4            20             1   \n",
       "4          17                    0            20           144   \n",
       "\n",
       "   33-Breast-Her2-ampl  42-Breast-Her2-ampl  Breast-454  48-Breast-WT  \\\n",
       "0                    0                    0           0             0   \n",
       "1                    0                    0           0             0   \n",
       "2                  108                   55           7            27   \n",
       "3                   26                   24           0             0   \n",
       "4                   26                   97           6            11   \n",
       "\n",
       "   50-Breast-WT  55-Breast-WT  Type-Unknown-6  Breast-564  59-Breast-WT  \\\n",
       "0             0             0               1           2             0   \n",
       "1             0             0               0           0             0   \n",
       "2             0             0              24          12           102   \n",
       "3             1             0               0           0             0   \n",
       "4             7            85             128          28           224   \n",
       "\n",
       "   Breast-590  Breast-60  Breast-603  ALK61-Breast-Her2-unknown  Breast-85  \\\n",
       "0           0          0           0                         23          0   \n",
       "1           0          0           0                          0          0   \n",
       "2          93          1          10                         41          4   \n",
       "3          20          0           0                         39          0   \n",
       "4         120         21          16                         20         21   \n",
       "\n",
       "   Breast-86  97-Breast-Her2-ampl  Breast-ALK-82  292-Liver-KRAS  \\\n",
       "0          1                    1              0              22   \n",
       "1          0                    0              0               0   \n",
       "2         16                    0              9              39   \n",
       "3          0                    0              7               0   \n",
       "4         20                   80             27              28   \n",
       "\n",
       "   316-Liver-KRAS  341-Liver-WT  376-Liver-WT  379-Liver-WT  Chol-410  \\\n",
       "0               0             8             0             0         0   \n",
       "1               0             0             0             0         0   \n",
       "2              13            19            61            17        18   \n",
       "3               0             0            10             0         9   \n",
       "4              37            25            43            23        11   \n",
       "\n",
       "   Chol-442  Chol-460  Chol-611  Chol-ALK-28  VU256-CRC  VU258-CRC  VU272-CRC  \\\n",
       "0         5         0         5            0          0          0          0   \n",
       "1         0         0         0            0          0          0          0   \n",
       "2        43         5        29           30          0          1          0   \n",
       "3         5         3         3            0          0          0          0   \n",
       "4        18        22        35           16         28          0          3   \n",
       "\n",
       "   VU276-CRC  Type-Unknown-1  329-CRC-KRAS  342-CRC-KRAS  Type-Unknown-5  \\\n",
       "0        241               0             0             0               0   \n",
       "1          0               0             0             0               0   \n",
       "2        180               8             0            29              14   \n",
       "3          0              13             0            27               1   \n",
       "4          0              37           112            12              21   \n",
       "\n",
       "   354-CRC-WT  356-CRC-WT  CRC-357  374-CRC-WT  378-CRC-WT  420-CRC-WT  \\\n",
       "0           0           0        5           0           1           4   \n",
       "1           0           0        0           0           0           0   \n",
       "2          28          20       18          18           0          39   \n",
       "3           0           0        7           0           5          10   \n",
       "4           5          11       18          89          14          20   \n",
       "\n",
       "   429-CRC-WT  CRC-450  453-CRC-KRAS  CRC-459  462-CRC-KRAS  CRC-463  \\\n",
       "0           0        0             0        2             0        0   \n",
       "1           0        0             0        0             0        0   \n",
       "2           4       20            14        8            63       17   \n",
       "3           0        4             0        0             0        3   \n",
       "4          93        1            47       12            32       16   \n",
       "\n",
       "   466-CRC-KRAS  474-CRC-WT  CRC-487  CRC-496  497-CRC-KRAS  504-CRC-WT  \\\n",
       "0             0           0        0        3             8           1   \n",
       "1             0           0        0        0             0           0   \n",
       "2            18           0        5       22            41          31   \n",
       "3             2          34        3        2            24           6   \n",
       "4            16          56       21       32            59          43   \n",
       "\n",
       "   CRC-511  CRC-523  CRC-524  527-CRC-WT  CRC-531  543-CRC-KRAS  546-CRC-KRAS  \\\n",
       "0        1        3        7           0        0            16             0   \n",
       "1        0        6        0           0        0             0             0   \n",
       "2       18        3       19          14        4            84            15   \n",
       "3        4        0        0          14        3             8             8   \n",
       "4        4        9       45          47       14            99            20   \n",
       "\n",
       "   548-CRC-KRAS  CRC-552  556-CRC-WT  CRC-565  CRC-576  VU280-GBM  \\\n",
       "0             0        4           5        0        0          0   \n",
       "1             0        0           0        0        0          0   \n",
       "2             6        7           2       21        0          1   \n",
       "3             0        1           0        0        0          0   \n",
       "4            28       31          34        7        9          0   \n",
       "\n",
       "   VU284-GBM-vIII  VU306-GBM-vIII  VU360-GBM-WT  VU372-GBM-vIII  VU378-GBM  \\\n",
       "0               4               0             0               0          7   \n",
       "1               0               0             0               0          0   \n",
       "2               0               0             0               0         50   \n",
       "3               0               0             0               0          5   \n",
       "4              13              21             1               9         18   \n",
       "\n",
       "   VU383Platelet-hiseq  VU393-GBM-vIII  VU394Platelet-hiseq  VU398-GBM  \\\n",
       "0                    0               0                    0          0   \n",
       "1                    0               0                    0          0   \n",
       "2                    0               0                    0          0   \n",
       "3                    0              14                    0          0   \n",
       "4                    0               6                    6          2   \n",
       "\n",
       "   VU402-2-GBM  VU406-2-GBM  VU408-GBM  VU417-GBM-vIII  VU418-GBM  VU419-GBM  \\\n",
       "0            0            4          0               0          0          0   \n",
       "1            0            0          0               0          0          0   \n",
       "2           21           22         31              95         15          0   \n",
       "3            0            0         23               0          0          0   \n",
       "4            7            5         11             100          0         20   \n",
       "\n",
       "   VU421-GBM  VU422-GBM  VU425-GBM  VU429-1-GBM-vIII  VU430-GBM  \\\n",
       "0          0          3          1                 0          3   \n",
       "1          0          0          0                 0          0   \n",
       "2          2          6         35                 0         12   \n",
       "3          3          0          8                 6         11   \n",
       "4         17         10         32                 1         36   \n",
       "\n",
       "   VU431-1-GBM-vIII  VU436-1-GBM-vIII  GBM-438  VU439-1-GBM-vIII  GBM-440  \\\n",
       "0                 0                 0        0                 0        0   \n",
       "1                 0                 0        0                 0        0   \n",
       "2                76                 6        5                27      105   \n",
       "3                38                 1        1                55       21   \n",
       "4                27                 1        3                 5       49   \n",
       "\n",
       "   VU443-1-GBM-WT  VU445-1-GBM-WT  VU454-1-GBM-WT  VU456-1-GBM-WT  \\\n",
       "0               3               4               8               0   \n",
       "1               0               0               0               0   \n",
       "2               9               0              75              16   \n",
       "3              10               0              32               6   \n",
       "4               0               7              29              12   \n",
       "\n",
       "   VU468-1-GBM-WT  GBM-471  GBM-475  GBM-476  GBM-480  GBM-484  GBM-485  \\\n",
       "0               0        3        9        2        0        0        0   \n",
       "1               0        0        0        0        0        0        0   \n",
       "2              19       19       12       30       54       37       33   \n",
       "3               3        0        0        7       17        1        9   \n",
       "4              33        0       13        2       31       12        3   \n",
       "\n",
       "   GBM-486  GBM-499  GBM-505  HD-1  HD-2-1  HD-3-1  HD-4  HD-5  HD-7  HD-8  \\\n",
       "0        0        0        0     0       0       9     0     0     0     0   \n",
       "1        0        0        0     0       0       0     0     0     0     0   \n",
       "2       56        4        0   100      27      22    25     2    42    47   \n",
       "3        2        0        0     6       3       2    30     4     2     0   \n",
       "4        4       10       11    11      13       2     2    23    14    12   \n",
       "\n",
       "   HD-9  HD-10  HD-11-1  HD-13-1  HD-14-1  HD-15-1  HD-16-1  HD-20-1  HD-22-1  \\\n",
       "0     0      8        8        0        3        0        8        0        5   \n",
       "1     0      0        0        0        0        0        0        0        0   \n",
       "2     0     88        6      146      128       32      129        7        5   \n",
       "3     0      1        8       39       64        7      132        0        9   \n",
       "4     1      0       49       60       52       10       95        6       49   \n",
       "\n",
       "   HD-34-1  HD-39-1  HD-40-1  HD-45-1  HD-46-1  Control-2  Control-3  \\\n",
       "0        0        1        0       14        0          0          0   \n",
       "1        0        0        0        0        0          0          0   \n",
       "2       12      131       34      184      214          0         75   \n",
       "3        3       41        0       97       70         17         23   \n",
       "4        8       56       26      168       44          4          6   \n",
       "\n",
       "   Control-4  Control-5  Control-6  Control-8  Control-9  Control-10  \\\n",
       "0          0          0          0          1          3           0   \n",
       "1          0          0          0          0          0           0   \n",
       "2          0         17        139          0        100          94   \n",
       "3          0          0          0         52         17           0   \n",
       "4          4         10          1         25         21           0   \n",
       "\n",
       "   Control-13  Type-Unknown-3  HD-17-1  HD-18-2  HD-21-1  HD-24-2  HD-25-2  \\\n",
       "0           0               5        0        0        0        0        0   \n",
       "1           0               0        0        0        0        0        0   \n",
       "2         190              22        2       10       16       18       16   \n",
       "3          12              25        0        0        3        0        0   \n",
       "4          27               4        4        9        6        5        4   \n",
       "\n",
       "   HD-26-2  HD-27-2  HD-29-2  HD-30-2  HD-31  HD-32-2  HD-33-1  HD-35-2  \\\n",
       "0        0        0        0        0      5        0        0        4   \n",
       "1        0        0        0        0      0        0        0        0   \n",
       "2       12       16        8        6     11        4       95       14   \n",
       "3       19        3        5        0      0        0       16        0   \n",
       "4       13        2       10        7     25        2       14       12   \n",
       "\n",
       "   HD-36  HD-37-2  HD-38-2  HD-44-2  HD-47-1  HD-48-1  HD-49-1  HD-50-1  \\\n",
       "0      0        0        0        4        0        0        0        0   \n",
       "1      0        0        0        0        0        0        0        0   \n",
       "2     11       33       49       83        4       24       30       23   \n",
       "3      5       11       13       16        1        8        4        4   \n",
       "4      2       17       19       24        0        1        5        2   \n",
       "\n",
       "   HD-51-1  HD-53-1  HD-54-1  VU274-CRC  297-Liver-WT  366-Liver-WT  \\\n",
       "0        0        0        0          0             0             0   \n",
       "1        0        0        0          0             0             0   \n",
       "2        9       15       15         56            29             8   \n",
       "3        2        3        9          0            11             0   \n",
       "4        5        0        3          5             5            33   \n",
       "\n",
       "   Lung-0082  Lung-0096  Lung-0001  Lung-0100  Lung-0101  Lung-0102  \\\n",
       "0          2          0          0          0          0          0   \n",
       "1          0          0          0          0          0          0   \n",
       "2         24         26        156         20          3         17   \n",
       "3          0          6          0          6          0          3   \n",
       "4          6          7          0         17          0         22   \n",
       "\n",
       "   Lung-0104  Lung-0110  Lung-0112  Lung-0119  Lung-004-EGFR  Lung-006  \\\n",
       "0          0          0          2          0              0         0   \n",
       "1          0          0          0          0              0         0   \n",
       "2          0         24         14         39             51         0   \n",
       "3          0          3         11          0             33         0   \n",
       "4         11         15         21          8             56         0   \n",
       "\n",
       "   Lung-0007  Lung-008  Lung-0012  Lung-0014J  Lung-0015  Lung-0016  \\\n",
       "0          0         0          0           0          0          0   \n",
       "1          0         0          0           0          0          0   \n",
       "2          0         0          0          32          0          0   \n",
       "3          0         0          0           0         44          0   \n",
       "4          4         0          0          16         45          0   \n",
       "\n",
       "   Lung-0017  Lung-0020J  Lung-0020J-1  Lung-0024J-1  Lung-0025J  Lung-0026J  \\\n",
       "0          0           5             0             0           0           0   \n",
       "1          0           0             0             0           0           0   \n",
       "2        196          46             1             3          12           0   \n",
       "3          6           6             0             0           0           0   \n",
       "4          0          21             1             1           4           1   \n",
       "\n",
       "   VU271-CRC  Lung-0029J-1-WT  Lung-0030J-1-WT  Lung-0037  Lung-0040  \\\n",
       "0          0                0                0         10          0   \n",
       "1          0                0                0          0          0   \n",
       "2          5                0                7        247         47   \n",
       "3          0                0                0          1          3   \n",
       "4         26               17                2          0         64   \n",
       "\n",
       "   Lung-0042  Lung-0046  Lung-0058-EGFR  Lung-0061-EGFR  Lung-0062-EGFR  \\\n",
       "0          0          0               0               7               0   \n",
       "1          0          0               0               0               0   \n",
       "2          0         50               0              18               0   \n",
       "3          0          0               5               0               0   \n",
       "4          6         13               5              13              31   \n",
       "\n",
       "   Lung-0074-EGFR  Lung-L-13  Lung-L-15  Lung-L-26  Lung-L-33  Lung-L-4  \\\n",
       "0               0         11          0          0          0         7   \n",
       "1               0          0          0          0          0         0   \n",
       "2              86         16         16         29         17        16   \n",
       "3              90          6          6          0          5        17   \n",
       "4              50         20         14         12         39         8   \n",
       "\n",
       "   VU260-CRC  286-Pancr-286-WT  314-Pancr-WT  327-Pancr-KRAS  328-Pancr-WT  \\\n",
       "0          0                 0             0               0             0   \n",
       "1          0                 0             0               0             0   \n",
       "2          0                60             0              35            85   \n",
       "3          0                17             7               0             0   \n",
       "4         24                22           107              21           139   \n",
       "\n",
       "   331-Pancr-WT  Panc-335  367-Pancr-KRAS  369-Pancr-KRAS  Panc-370  \\\n",
       "0             0         3               0               0         0   \n",
       "1             0         0               0               0         0   \n",
       "2            57        19               0              17        25   \n",
       "3             0         8              17               0         0   \n",
       "4           125        43              22             114        31   \n",
       "\n",
       "   371-Pancr-KRAS  380-Pancr-KRAS  382-Pancr-KRAS  387-Pancr-WT  388-Pancr-WT  \\\n",
       "0               0               0               0             0             8   \n",
       "1               0               0               0             0             0   \n",
       "2              28               0              13            14            29   \n",
       "3               0               4               0             8             0   \n",
       "4              16              82              40            14            65   \n",
       "\n",
       "   389-Pancr-WT  390-Pancr-KRAS  391-Pancr-KRAS  431-Pancr-WT  434-Pancr-WT  \\\n",
       "0            11               0               0             0             0   \n",
       "1             0               0               0             0             0   \n",
       "2            16              27               0             7             0   \n",
       "3             0               3               0             0             1   \n",
       "4            25              33              10            37            34   \n",
       "\n",
       "   Panc-435  448-Pancr-KRAS  Type-Unknown-4  Panc-451  Panc-461  Panc-477  \\\n",
       "0         0              33               0         8         0         0   \n",
       "1         0               0               0         0         0         0   \n",
       "2         0              42              13        16        13         4   \n",
       "3         6               5               0         0         9         6   \n",
       "4        20              37              34        31        14         4   \n",
       "\n",
       "   488-Pancr-WT  Panc-493  498-Pancr-WT  Panc-501  Panc-503  512-Pancr-KRAS  \\\n",
       "0             0         0             0         0         0               4   \n",
       "1             0         0             0         0         0               0   \n",
       "2            39        20            79        18         0              31   \n",
       "3             0         0             0         0         2              17   \n",
       "4             0         3            12        32         8              37   \n",
       "\n",
       "   553-Pancr-KRAS  Panc-593  Panc-597  MGH-BrCa-H-11  MGH-BrCa-H-59  \\\n",
       "0               0         1         6              2              0   \n",
       "1               0         0         0              0              0   \n",
       "2              92         6        18             21              0   \n",
       "3               0         0         2             11              0   \n",
       "4               8         6         9             38              3   \n",
       "\n",
       "   MGH-BrCa-H-66  MGH-BrCa-H-68  MGH-BrCa-H-74  MGH-BrCa-H-75  MGH-CRC-368  \\\n",
       "0              7              0              1              3            0   \n",
       "1              0              0              0              0            0   \n",
       "2             25             11              2             30           17   \n",
       "3              0              5              4              3            2   \n",
       "4              5              6             19             15            5   \n",
       "\n",
       "   Vumc-NSCLC-49  Vumc-NSCLC-57  Vumc-NSCLC-59  MGH-BrCa-H31-TR-489  \\\n",
       "0              0              9              0                    3   \n",
       "1              0              0              0                    0   \n",
       "2             14             25             40                   12   \n",
       "3              4              1             24                    2   \n",
       "4             13             15             37                   41   \n",
       "\n",
       "   MGH-BrCa-H38-TR467  MGH-BrCa-H63-TR-491  MGH-BrCa-H76-TR469  \\\n",
       "0                   0                    0                   0   \n",
       "1                   0                    0                   0   \n",
       "2                   2                   25                  18   \n",
       "3                   0                   10                  13   \n",
       "4                   1                   17                  11   \n",
       "\n",
       "   MGH-BrCa-H81-TR492  MGH-BrCa-H83-TR550  MGH-BrCa-H84-TR551  \\\n",
       "0                   0                   0                   0   \n",
       "1                   0                   0                   0   \n",
       "2                  40                   6                  32   \n",
       "3                   9                   0                   3   \n",
       "4                  79                  16                   8   \n",
       "\n",
       "   MGH-BrCa-H89-TR471  MGH-BrCa-H92-TR472  MGH-CRC-412-TR466  \\\n",
       "0                   0                   0                  0   \n",
       "1                   0                   0                  0   \n",
       "2                  12                   4                  0   \n",
       "3                   0                   0                  0   \n",
       "4                   1                  36                  0   \n",
       "\n",
       "   MGH-CRC-BRAF4-TR547  MGH-CRC-BRAF5-TR548  MGH-CRC-BRAF6-TR465  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    3                   24                   12   \n",
       "3                   56                    0                    6   \n",
       "4                   17                    1                   48   \n",
       "\n",
       "   MGH-NSCLC-L01-TR458  MGH-NSCLC-L03-TR488  MGH-NSCLC-L07-TR459  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                   31                   27   \n",
       "3                   21                    0                    0   \n",
       "4                   18                   23                   11   \n",
       "\n",
       "   MGH-NSCLC-L11-TR477  MGH-NSCLC-L12-TR478  MGH-NSCLC-L18-TR479  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                   46                   42                    6   \n",
       "3                   20                    0                    0   \n",
       "4                   12                    3                    9   \n",
       "\n",
       "   MGH-NSCLC-L19-TR461  MGH-NSCLC-L20-TR500  MGH-NSCLC-L22-TR462  \\\n",
       "0                    0                    0                    5   \n",
       "1                    0                    0                    0   \n",
       "2                   30                    2                   41   \n",
       "3                    0                    0                    0   \n",
       "4                   40                   23                    1   \n",
       "\n",
       "   MGH-NSCLC-L23-TR524  MGH-NSCLC-L25-TR480  MGH-NSCLC-L39-TR519  \\\n",
       "0                   10                    9                    0   \n",
       "1                    0                    0                    0   \n",
       "2                   40                   15                    1   \n",
       "3                    0                    0                    0   \n",
       "4                    6                   22                   30   \n",
       "\n",
       "   MGH-NSCLC-L40-TR520  MGH-NSCLC-L51-TR521  MGH-NSCLC-L58-TR525  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "2                    0                   13                    0   \n",
       "3                    0                    0                    0   \n",
       "4                    0                    1                   50   \n",
       "\n",
       "   MGH-NSCLC-L59-TR522  MGH-NSCLC-L65-TR523  \n",
       "0                    0                    0  \n",
       "1                    0                    0  \n",
       "2                    0                   10  \n",
       "3                    0                    0  \n",
       "4                    6                    2  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tep_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "879f3b58-d79e-401c-af0b-73f9b15d9f76",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0d34d0ab-3014-4f5c-a20f-38f2ae1db8af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "!Sample_geo_accession\n",
      "!Sample_status\n",
      "!Sample_submission_date\n",
      "!Sample_last_update_date\n",
      "!Sample_type\n",
      "!Sample_channel_count\n",
      "!Sample_source_name_ch1\n",
      "!Sample_organism_ch1\n",
      "!Sample_characteristics_ch1\n",
      "!Sample_characteristics_ch1.1\n",
      "!Sample_characteristics_ch1.2\n",
      "!Sample_characteristics_ch1.3\n",
      "!Sample_characteristics_ch1.4\n",
      "!Sample_characteristics_ch1.5\n",
      "!Sample_molecule_ch1\n",
      "!Sample_extract_protocol_ch1\n",
      "!Sample_extract_protocol_ch1.1\n",
      "!Sample_taxid_ch1\n",
      "!Sample_description\n",
      "!Sample_data_processing\n",
      "!Sample_data_processing.1\n",
      "!Sample_data_processing.2\n",
      "!Sample_data_processing.3\n",
      "!Sample_data_processing.4\n",
      "!Sample_data_processing.5\n",
      "!Sample_platform_id\n",
      "!Sample_contact_name\n",
      "!Sample_contact_email\n",
      "!Sample_contact_laboratory\n",
      "!Sample_contact_department\n",
      "!Sample_contact_institute\n",
      "!Sample_contact_address\n",
      "!Sample_contact_city\n",
      "!Sample_contact_zip/postal_code\n",
      "!Sample_contact_country\n",
      "!Sample_data_row_count\n",
      "!Sample_instrument_model\n",
      "!Sample_library_selection\n",
      "!Sample_library_source\n",
      "!Sample_library_strategy\n",
      "!Sample_relation\n",
      "!Sample_relation.1\n",
      "!Sample_supplementary_file_1\n",
      "!series_matrix_table_begin\n",
      "\"ID_REF\"\n",
      "!series_matrix_table_end\n"
     ]
    }
   ],
   "source": [
    "series_col_list = series_df.columns.to_list()\n",
    "\n",
    "for i in series_col_list:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "88eae2ff-efe3-48e6-abdf-19d8a7b8fb82",
   "metadata": {},
   "outputs": [],
   "source": [
    "#series_df.sort_values(by=\"\", axis=0, ascending=True, inplace=False, kind='quicksort', na_position='last', ignore_index=False, key=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "535d1104-d8a6-447b-9d13-e2a2b5ead650",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unnamed: 0\n",
      "3-Breast-Her2-ampl\n",
      "8-Breast-WT\n",
      "10-Breast-Her2-ampl\n",
      "Breast-100\n",
      "15-Breast-Her2-ampl\n",
      "16-Breast-WT\n",
      "21-Breast-WT\n",
      "33-Breast-Her2-ampl\n",
      "42-Breast-Her2-ampl\n",
      "Breast-454\n",
      "48-Breast-WT\n",
      "50-Breast-WT\n",
      "55-Breast-WT\n",
      "Type-Unknown-6\n",
      "Breast-564\n",
      "59-Breast-WT\n",
      "Breast-590\n",
      "Breast-60\n",
      "Breast-603\n",
      "ALK61-Breast-Her2-unknown\n",
      "Breast-85\n",
      "Breast-86\n",
      "97-Breast-Her2-ampl\n",
      "Breast-ALK-82\n",
      "292-Liver-KRAS\n",
      "316-Liver-KRAS\n",
      "341-Liver-WT\n",
      "376-Liver-WT\n",
      "379-Liver-WT\n",
      "Chol-410\n",
      "Chol-442\n",
      "Chol-460\n",
      "Chol-611\n",
      "Chol-ALK-28\n",
      "VU256-CRC\n",
      "VU258-CRC\n",
      "VU272-CRC\n",
      "VU276-CRC\n",
      "Type-Unknown-1\n",
      "329-CRC-KRAS\n",
      "342-CRC-KRAS\n",
      "Type-Unknown-5\n",
      "354-CRC-WT\n",
      "356-CRC-WT\n",
      "CRC-357\n",
      "374-CRC-WT\n",
      "378-CRC-WT\n",
      "420-CRC-WT\n",
      "429-CRC-WT\n",
      "CRC-450\n",
      "453-CRC-KRAS\n",
      "CRC-459\n",
      "462-CRC-KRAS\n",
      "CRC-463\n",
      "466-CRC-KRAS\n",
      "474-CRC-WT\n",
      "CRC-487\n",
      "CRC-496\n",
      "497-CRC-KRAS\n",
      "504-CRC-WT\n",
      "CRC-511\n",
      "CRC-523\n",
      "CRC-524\n",
      "527-CRC-WT\n",
      "CRC-531\n",
      "543-CRC-KRAS\n",
      "546-CRC-KRAS\n",
      "548-CRC-KRAS\n",
      "CRC-552\n",
      "556-CRC-WT\n",
      "CRC-565\n",
      "CRC-576\n",
      "VU280-GBM\n",
      "VU284-GBM-vIII\n",
      "VU306-GBM-vIII\n",
      "VU360-GBM-WT\n",
      "VU372-GBM-vIII\n",
      "VU378-GBM\n",
      "VU383Platelet-hiseq\n",
      "VU393-GBM-vIII\n",
      "VU394Platelet-hiseq\n",
      "VU398-GBM\n",
      "VU402-2-GBM\n",
      "VU406-2-GBM\n",
      "VU408-GBM\n",
      "VU417-GBM-vIII\n",
      "VU418-GBM\n",
      "VU419-GBM\n",
      "VU421-GBM\n",
      "VU422-GBM\n",
      "VU425-GBM\n",
      "VU429-1-GBM-vIII\n",
      "VU430-GBM\n",
      "VU431-1-GBM-vIII\n",
      "VU436-1-GBM-vIII\n",
      "GBM-438\n",
      "VU439-1-GBM-vIII\n",
      "GBM-440\n",
      "VU443-1-GBM-WT\n",
      "VU445-1-GBM-WT\n",
      "VU454-1-GBM-WT\n",
      "VU456-1-GBM-WT\n",
      "VU468-1-GBM-WT\n",
      "GBM-471\n",
      "GBM-475\n",
      "GBM-476\n",
      "GBM-480\n",
      "GBM-484\n",
      "GBM-485\n",
      "GBM-486\n",
      "GBM-499\n",
      "GBM-505\n",
      "HD-1\n",
      "HD-2-1\n",
      "HD-3-1\n",
      "HD-4\n",
      "HD-5\n",
      "HD-7\n",
      "HD-8\n",
      "HD-9\n",
      "HD-10\n",
      "HD-11-1\n",
      "HD-13-1\n",
      "HD-14-1\n",
      "HD-15-1\n",
      "HD-16-1\n",
      "HD-20-1\n",
      "HD-22-1\n",
      "HD-34-1\n",
      "HD-39-1\n",
      "HD-40-1\n",
      "HD-45-1\n",
      "HD-46-1\n",
      "Control-2\n",
      "Control-3\n",
      "Control-4\n",
      "Control-5\n",
      "Control-6\n",
      "Control-8\n",
      "Control-9\n",
      "Control-10\n",
      "Control-13\n",
      "Type-Unknown-3\n",
      "HD-17-1\n",
      "HD-18-2\n",
      "HD-21-1\n",
      "HD-24-2\n",
      "HD-25-2\n",
      "HD-26-2\n",
      "HD-27-2\n",
      "HD-29-2\n",
      "HD-30-2\n",
      "HD-31\n",
      "HD-32-2\n",
      "HD-33-1\n",
      "HD-35-2\n",
      "HD-36\n",
      "HD-37-2\n",
      "HD-38-2\n",
      "HD-44-2\n",
      "HD-47-1\n",
      "HD-48-1\n",
      "HD-49-1\n",
      "HD-50-1\n",
      "HD-51-1\n",
      "HD-53-1\n",
      "HD-54-1\n",
      "VU274-CRC\n",
      "297-Liver-WT\n",
      "366-Liver-WT\n",
      "Lung-0082\n",
      "Lung-0096\n",
      "Lung-0001\n",
      "Lung-0100\n",
      "Lung-0101\n",
      "Lung-0102\n",
      "Lung-0104\n",
      "Lung-0110\n",
      "Lung-0112\n",
      "Lung-0119\n",
      "Lung-004-EGFR\n",
      "Lung-006\n",
      "Lung-0007\n",
      "Lung-008\n",
      "Lung-0012\n",
      "Lung-0014J\n",
      "Lung-0015\n",
      "Lung-0016\n",
      "Lung-0017\n",
      "Lung-0020J\n",
      "Lung-0020J-1\n",
      "Lung-0024J-1\n",
      "Lung-0025J\n",
      "Lung-0026J\n",
      "VU271-CRC\n",
      "Lung-0029J-1-WT\n",
      "Lung-0030J-1-WT\n",
      "Lung-0037\n",
      "Lung-0040\n",
      "Lung-0042\n",
      "Lung-0046\n",
      "Lung-0058-EGFR\n",
      "Lung-0061-EGFR\n",
      "Lung-0062-EGFR\n",
      "Lung-0074-EGFR\n",
      "Lung-L-13\n",
      "Lung-L-15\n",
      "Lung-L-26\n",
      "Lung-L-33\n",
      "Lung-L-4\n",
      "VU260-CRC\n",
      "286-Pancr-286-WT\n",
      "314-Pancr-WT\n",
      "327-Pancr-KRAS\n",
      "328-Pancr-WT\n",
      "331-Pancr-WT\n",
      "Panc-335\n",
      "367-Pancr-KRAS\n",
      "369-Pancr-KRAS\n",
      "Panc-370\n",
      "371-Pancr-KRAS\n",
      "380-Pancr-KRAS\n",
      "382-Pancr-KRAS\n",
      "387-Pancr-WT\n",
      "388-Pancr-WT\n",
      "389-Pancr-WT\n",
      "390-Pancr-KRAS\n",
      "391-Pancr-KRAS\n",
      "431-Pancr-WT\n",
      "434-Pancr-WT\n",
      "Panc-435\n",
      "448-Pancr-KRAS\n",
      "Type-Unknown-4\n",
      "Panc-451\n",
      "Panc-461\n",
      "Panc-477\n",
      "488-Pancr-WT\n",
      "Panc-493\n",
      "498-Pancr-WT\n",
      "Panc-501\n",
      "Panc-503\n",
      "512-Pancr-KRAS\n",
      "553-Pancr-KRAS\n",
      "Panc-593\n",
      "Panc-597\n",
      "MGH-BrCa-H-11\n",
      "MGH-BrCa-H-59\n",
      "MGH-BrCa-H-66\n",
      "MGH-BrCa-H-68\n",
      "MGH-BrCa-H-74\n",
      "MGH-BrCa-H-75\n",
      "MGH-CRC-368\n",
      "Vumc-NSCLC-49\n",
      "Vumc-NSCLC-57\n",
      "Vumc-NSCLC-59\n",
      "MGH-BrCa-H31-TR-489\n",
      "MGH-BrCa-H38-TR467\n",
      "MGH-BrCa-H63-TR-491\n",
      "MGH-BrCa-H76-TR469\n",
      "MGH-BrCa-H81-TR492\n",
      "MGH-BrCa-H83-TR550\n",
      "MGH-BrCa-H84-TR551\n",
      "MGH-BrCa-H89-TR471\n",
      "MGH-BrCa-H92-TR472\n",
      "MGH-CRC-412-TR466\n",
      "MGH-CRC-BRAF4-TR547\n",
      "MGH-CRC-BRAF5-TR548\n",
      "MGH-CRC-BRAF6-TR465\n",
      "MGH-NSCLC-L01-TR458\n",
      "MGH-NSCLC-L03-TR488\n",
      "MGH-NSCLC-L07-TR459\n",
      "MGH-NSCLC-L11-TR477\n",
      "MGH-NSCLC-L12-TR478\n",
      "MGH-NSCLC-L18-TR479\n",
      "MGH-NSCLC-L19-TR461\n",
      "MGH-NSCLC-L20-TR500\n",
      "MGH-NSCLC-L22-TR462\n",
      "MGH-NSCLC-L23-TR524\n",
      "MGH-NSCLC-L25-TR480\n",
      "MGH-NSCLC-L39-TR519\n",
      "MGH-NSCLC-L40-TR520\n",
      "MGH-NSCLC-L51-TR521\n",
      "MGH-NSCLC-L58-TR525\n",
      "MGH-NSCLC-L59-TR522\n",
      "MGH-NSCLC-L65-TR523\n"
     ]
    }
   ],
   "source": [
    "col_list = tep_df.columns.to_list()\n",
    "\n",
    "for i in col_list:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "412865b3-a4fa-44c7-a2b1-541cd0c76b65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"GSM1662534\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"3-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-03\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662534\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"GSM1662535\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"8-Breast-WT\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-08\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662535\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"GSM1662536\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"10-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-10\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662536\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"GSM1662537\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"Breast-100\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-100\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch04\"</td>\n",
       "      <td>\"mutational subclass: Triple Negative\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662537\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"GSM1662538\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"15-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-15\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662538\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  !Sample_geo_accession           !Sample_status !Sample_submission_date  \\\n",
       "0          \"GSM1662534\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "1          \"GSM1662535\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "2          \"GSM1662536\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "3          \"GSM1662537\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "4          \"GSM1662538\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "\n",
       "  !Sample_last_update_date !Sample_type !Sample_channel_count  \\\n",
       "0            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "1            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "2            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "3            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "4            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "\n",
       "  !Sample_source_name_ch1 !Sample_organism_ch1 !Sample_characteristics_ch1  \\\n",
       "0    \"3-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "1           \"8-Breast-WT\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "2   \"10-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "3            \"Breast-100\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "4   \"15-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.1 !Sample_characteristics_ch1.2  \\\n",
       "0     \"cell type: Thrombocytes\"       \"patient id: Breast-03\"   \n",
       "1     \"cell type: Thrombocytes\"       \"patient id: Breast-08\"   \n",
       "2     \"cell type: Thrombocytes\"       \"patient id: Breast-10\"   \n",
       "3     \"cell type: Thrombocytes\"      \"patient id: Breast-100\"   \n",
       "4     \"cell type: Thrombocytes\"       \"patient id: Breast-15\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.3 !Sample_characteristics_ch1.4  \\\n",
       "0         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "1         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "2         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "3         \"cancer type: Breast\"              \"batch: Batch04\"   \n",
       "4         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "\n",
       "            !Sample_characteristics_ch1.5 !Sample_molecule_ch1  \\\n",
       "0            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "1               \"mutational subclass: wt\"          \"total RNA\"   \n",
       "2            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "3  \"mutational subclass: Triple Negative\"          \"total RNA\"   \n",
       "4            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "\n",
       "                        !Sample_extract_protocol_ch1  \\\n",
       "0  \"Blood platelets were isolated from whole bloo...   \n",
       "1  \"Blood platelets were isolated from whole bloo...   \n",
       "2  \"Blood platelets were isolated from whole bloo...   \n",
       "3  \"Blood platelets were isolated from whole bloo...   \n",
       "4  \"Blood platelets were isolated from whole bloo...   \n",
       "\n",
       "                      !Sample_extract_protocol_ch1.1 !Sample_taxid_ch1  \\\n",
       "0  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "1  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "2  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "3  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "4  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "\n",
       "  !Sample_description                            !Sample_data_processing  \\\n",
       "0                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "1                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "2                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "3                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "4                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "\n",
       "                           !Sample_data_processing.1  \\\n",
       "0  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "1  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "2  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "3  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "4  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "\n",
       "                           !Sample_data_processing.2  \\\n",
       "0  \"Selection of intron-spanning reads was perfor...   \n",
       "1  \"Selection of intron-spanning reads was perfor...   \n",
       "2  \"Selection of intron-spanning reads was perfor...   \n",
       "3  \"Selection of intron-spanning reads was perfor...   \n",
       "4  \"Selection of intron-spanning reads was perfor...   \n",
       "\n",
       "                           !Sample_data_processing.3  \\\n",
       "0  \"Read summarization was performed using HTSeq ...   \n",
       "1  \"Read summarization was performed using HTSeq ...   \n",
       "2  \"Read summarization was performed using HTSeq ...   \n",
       "3  \"Read summarization was performed using HTSeq ...   \n",
       "4  \"Read summarization was performed using HTSeq ...   \n",
       "\n",
       "  !Sample_data_processing.4  \\\n",
       "0      \"Genome_build: hg19\"   \n",
       "1      \"Genome_build: hg19\"   \n",
       "2      \"Genome_build: hg19\"   \n",
       "3      \"Genome_build: hg19\"   \n",
       "4      \"Genome_build: hg19\"   \n",
       "\n",
       "                           !Sample_data_processing.5 !Sample_platform_id  \\\n",
       "0  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "1  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "2  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "3  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "4  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "\n",
       "  !Sample_contact_name !Sample_contact_email       !Sample_contact_laboratory  \\\n",
       "0        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "1        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "2        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "3        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "4        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "\n",
       "  !Sample_contact_department       !Sample_contact_institute  \\\n",
       "0             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "1             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "2             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "3             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "4             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "\n",
       "  !Sample_contact_address !Sample_contact_city  \\\n",
       "0     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "1     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "2     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "3     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "4     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "\n",
       "  !Sample_contact_zip/postal_code !Sample_contact_country  \\\n",
       "0                       \"1081 HV\"           \"Netherlands\"   \n",
       "1                       \"1081 HV\"           \"Netherlands\"   \n",
       "2                       \"1081 HV\"           \"Netherlands\"   \n",
       "3                       \"1081 HV\"           \"Netherlands\"   \n",
       "4                       \"1081 HV\"           \"Netherlands\"   \n",
       "\n",
       "  !Sample_data_row_count !Sample_instrument_model !Sample_library_selection  \\\n",
       "0                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "1                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "2                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "3                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "4                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "\n",
       "  !Sample_library_source !Sample_library_strategy  \\\n",
       "0       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "1       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "2       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "3       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "4       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "\n",
       "                                    !Sample_relation  \\\n",
       "0  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "1  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "2  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "3  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "4  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "\n",
       "                                  !Sample_relation.1  \\\n",
       "0  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "1  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "2  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "3  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "4  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "\n",
       "  !Sample_supplementary_file_1  !series_matrix_table_begin      \"ID_REF\"  \\\n",
       "0                       \"NONE\"                         NaN  \"GSM1662534\"   \n",
       "1                       \"NONE\"                         NaN  \"GSM1662535\"   \n",
       "2                       \"NONE\"                         NaN  \"GSM1662536\"   \n",
       "3                       \"NONE\"                         NaN  \"GSM1662537\"   \n",
       "4                       \"NONE\"                         NaN  \"GSM1662538\"   \n",
       "\n",
       "   !series_matrix_table_end  \n",
       "0                       NaN  \n",
       "1                       NaN  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                       NaN  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "\n",
    "series_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "95e14e5e-c17b-41c0-aaa8-19a8c3899f42",
   "metadata": {},
   "outputs": [],
   "source": [
    "# columns for preprocessing: ch1.3, ch1.2, ch1.1, ch1.4, ch1.5, ch1\n",
    "# fixed: ch1, 1.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "95977535-effe-4656-af99-77f743280d39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"cancer type: Breast\"' '\"cancer type: Hepatobiliary\"'\n",
      " '\"cancer type: CRC\"' '\"cancer type: GBM\"' '\"cancer type: HC\"'\n",
      " '\"cancer type: Lung\"' '\"cancer type: Pancreas\"'\n",
      " '\"mutational subclass: PIK3CA\"' '\"mutational subclass: wt\"'\n",
      " '\"mutational subclass: HER2+\"' '\"mutational subclass: KRAS\"'\n",
      " '\"mutational subclass: EGFR\"']\n"
     ]
    }
   ],
   "source": [
    "unique_cancer_types = series_df['!Sample_characteristics_ch1.3'].unique()\n",
    "print(unique_cancer_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "02f07b89-b990-4800-95b3-5b980aa6ecbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [!Sample_geo_accession, !Sample_status, !Sample_submission_date, !Sample_last_update_date, !Sample_type, !Sample_channel_count, !Sample_source_name_ch1, !Sample_organism_ch1, !Sample_characteristics_ch1, !Sample_characteristics_ch1.1, !Sample_characteristics_ch1.2, !Sample_characteristics_ch1.3, !Sample_characteristics_ch1.4, !Sample_characteristics_ch1.5, !Sample_molecule_ch1, !Sample_extract_protocol_ch1, !Sample_extract_protocol_ch1.1, !Sample_taxid_ch1, !Sample_description, !Sample_data_processing, !Sample_data_processing.1, !Sample_data_processing.2, !Sample_data_processing.3, !Sample_data_processing.4, !Sample_data_processing.5, !Sample_platform_id, !Sample_contact_name, !Sample_contact_email, !Sample_contact_laboratory, !Sample_contact_department, !Sample_contact_institute, !Sample_contact_address, !Sample_contact_city, !Sample_contact_zip/postal_code, !Sample_contact_country, !Sample_data_row_count, !Sample_instrument_model, !Sample_library_selection, !Sample_library_source, !Sample_library_strategy, !Sample_relation, !Sample_relation.1, !Sample_supplementary_file_1, !series_matrix_table_begin, \"ID_REF\", !series_matrix_table_end]\n",
       "Index: []"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fixing 1.3\n",
    "\n",
    "filtered_df =  series_df[ series_df['!Sample_characteristics_ch1'].str.contains(\"mutational subclass:\", case=False, na=False)]\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "filtered_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "e5f3b2eb-b4db-4ba4-bb31-7a399c23e012",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Boolean mask for rows where column contains substring\n",
    "\n",
    "mask = series_df['!Sample_characteristics_ch1.3'].str.contains(\"mutational subclass:\", case=False, na=False)\n",
    "\n",
    "# Swap values only for those rows\n",
    "series_df.loc[mask, ['!Sample_characteristics_ch1.3', '!Sample_characteristics_ch1.5']] = \\\n",
    "    series_df.loc[mask, ['!Sample_characteristics_ch1.5', '!Sample_characteristics_ch1.3']].values\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "4f4dc0c5-b1df-4d7c-8715-aad5cf64fa6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>\"GSM1817093\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H-11\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H11\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch05\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: PIK3CA\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817093\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>\"GSM1817094\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H31-TR-489\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H31\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817094\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>\"GSM1817095\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H38-TR467\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H38\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817095\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>\"GSM1817096\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H-59\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H59\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch05\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817096\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>\"GSM1817097\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H63-TR-491\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H63\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817097\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    !Sample_geo_accession           !Sample_status !Sample_submission_date  \\\n",
       "245          \"GSM1817093\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "246          \"GSM1817094\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "247          \"GSM1817095\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "248          \"GSM1817096\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "249          \"GSM1817097\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "\n",
       "    !Sample_last_update_date !Sample_type !Sample_channel_count  \\\n",
       "245            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "246            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "247            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "248            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "249            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "\n",
       "    !Sample_source_name_ch1 !Sample_organism_ch1 !Sample_characteristics_ch1  \\\n",
       "245         \"MGH-BrCa-H-11\"       \"Homo sapiens\"    \"patient id: Breast-H11\"   \n",
       "246   \"MGH-BrCa-H31-TR-489\"       \"Homo sapiens\"    \"patient id: Breast-H31\"   \n",
       "247    \"MGH-BrCa-H38-TR467\"       \"Homo sapiens\"    \"patient id: Breast-H38\"   \n",
       "248         \"MGH-BrCa-H-59\"       \"Homo sapiens\"    \"patient id: Breast-H59\"   \n",
       "249   \"MGH-BrCa-H63-TR-491\"       \"Homo sapiens\"    \"patient id: Breast-H63\"   \n",
       "\n",
       "    !Sample_characteristics_ch1.1 !Sample_characteristics_ch1.2  \\\n",
       "245         \"cancer type: Breast\"              \"batch: Batch05\"   \n",
       "246         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "247         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "248         \"cancer type: Breast\"              \"batch: Batch05\"   \n",
       "249         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "\n",
       "    !Sample_characteristics_ch1.3 !Sample_characteristics_ch1.4  \\\n",
       "245     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "246     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "247     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "248     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "249     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "\n",
       "     !Sample_characteristics_ch1.5 !Sample_molecule_ch1  \\\n",
       "245  \"mutational subclass: PIK3CA\"          \"total RNA\"   \n",
       "246      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "247      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "248      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "249      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "\n",
       "                          !Sample_extract_protocol_ch1  \\\n",
       "245  \"Blood platelets were isolated from whole bloo...   \n",
       "246  \"Blood platelets were isolated from whole bloo...   \n",
       "247  \"Blood platelets were isolated from whole bloo...   \n",
       "248  \"Blood platelets were isolated from whole bloo...   \n",
       "249  \"Blood platelets were isolated from whole bloo...   \n",
       "\n",
       "                        !Sample_extract_protocol_ch1.1 !Sample_taxid_ch1  \\\n",
       "245  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "246  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "247  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "248  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "249  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "\n",
       "    !Sample_description                            !Sample_data_processing  \\\n",
       "245                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "246                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "247                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "248                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "249                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "\n",
       "                             !Sample_data_processing.1  \\\n",
       "245  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "246  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "247  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "248  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "249  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "\n",
       "                             !Sample_data_processing.2  \\\n",
       "245  \"Selection of intron-spanning reads was perfor...   \n",
       "246  \"Selection of intron-spanning reads was perfor...   \n",
       "247  \"Selection of intron-spanning reads was perfor...   \n",
       "248  \"Selection of intron-spanning reads was perfor...   \n",
       "249  \"Selection of intron-spanning reads was perfor...   \n",
       "\n",
       "                             !Sample_data_processing.3  \\\n",
       "245  \"Read summarization was performed using HTSeq ...   \n",
       "246  \"Read summarization was performed using HTSeq ...   \n",
       "247  \"Read summarization was performed using HTSeq ...   \n",
       "248  \"Read summarization was performed using HTSeq ...   \n",
       "249  \"Read summarization was performed using HTSeq ...   \n",
       "\n",
       "    !Sample_data_processing.4  \\\n",
       "245      \"Genome_build: hg19\"   \n",
       "246      \"Genome_build: hg19\"   \n",
       "247      \"Genome_build: hg19\"   \n",
       "248      \"Genome_build: hg19\"   \n",
       "249      \"Genome_build: hg19\"   \n",
       "\n",
       "                             !Sample_data_processing.5 !Sample_platform_id  \\\n",
       "245  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "246  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "247  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "248  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "249  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "\n",
       "    !Sample_contact_name !Sample_contact_email  \\\n",
       "245        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "246        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "247        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "248        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "249        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "\n",
       "          !Sample_contact_laboratory !Sample_contact_department  \\\n",
       "245  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "246  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "247  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "248  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "249  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "\n",
       "          !Sample_contact_institute !Sample_contact_address  \\\n",
       "245  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "246  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "247  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "248  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "249  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "\n",
       "    !Sample_contact_city !Sample_contact_zip/postal_code  \\\n",
       "245          \"Amsterdam\"                       \"1081 HV\"   \n",
       "246          \"Amsterdam\"                       \"1081 HV\"   \n",
       "247          \"Amsterdam\"                       \"1081 HV\"   \n",
       "248          \"Amsterdam\"                       \"1081 HV\"   \n",
       "249          \"Amsterdam\"                       \"1081 HV\"   \n",
       "\n",
       "    !Sample_contact_country !Sample_data_row_count !Sample_instrument_model  \\\n",
       "245           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "246           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "247           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "248           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "249           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "\n",
       "    !Sample_library_selection !Sample_library_source !Sample_library_strategy  \\\n",
       "245                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "246                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "247                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "248                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "249                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "\n",
       "                                      !Sample_relation  \\\n",
       "245  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "246  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "247  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "248  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "249  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "\n",
       "                                    !Sample_relation.1  \\\n",
       "245  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "246  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "247  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "248  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "249  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "\n",
       "    !Sample_supplementary_file_1  !series_matrix_table_begin      \"ID_REF\"  \\\n",
       "245                       \"NONE\"                         NaN  \"GSM1817093\"   \n",
       "246                       \"NONE\"                         NaN  \"GSM1817094\"   \n",
       "247                       \"NONE\"                         NaN  \"GSM1817095\"   \n",
       "248                       \"NONE\"                         NaN  \"GSM1817096\"   \n",
       "249                       \"NONE\"                         NaN  \"GSM1817097\"   \n",
       "\n",
       "     !series_matrix_table_end  \n",
       "245                       NaN  \n",
       "246                       NaN  \n",
       "247                       NaN  \n",
       "248                       NaN  \n",
       "249                       NaN  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fixing 1 (with 1.4)\n",
    "\n",
    "filtered_df =  series_df[ series_df['!Sample_characteristics_ch1'].str.contains(\"patient\", case=False, na=False)]\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "filtered_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5d7454a7-fa59-4933-a0ba-8a11cf5656bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"patient id: Breast-03\"' '\"patient id: Breast-08\"'\n",
      " '\"patient id: Breast-10\"' '\"patient id: Breast-100\"'\n",
      " '\"patient id: Breast-15\"' '\"patient id: Breast-16\"'\n",
      " '\"patient id: Breast-21\"' '\"patient id: Breast-33\"'\n",
      " '\"patient id: Breast-42\"' '\"patient id: Breast-454\"'\n",
      " '\"patient id: Breast-48\"' '\"patient id: Breast-50\"'\n",
      " '\"patient id: Breast-55\"' '\"patient id: Breast-56\"'\n",
      " '\"patient id: Breast-564\"' '\"patient id: Breast-59\"'\n",
      " '\"patient id: Breast-590\"' '\"patient id: Breast-60\"'\n",
      " '\"patient id: Breast-603\"' '\"patient id: Breast-61\"'\n",
      " '\"patient id: Breast-85\"' '\"patient id: Breast-86\"'\n",
      " '\"patient id: Breast-97\"' '\"patient id: Breast-ALK-82\"'\n",
      " '\"patient id: Chol-292\"' '\"patient id: Chol-316\"'\n",
      " '\"patient id: Chol-341\"' '\"patient id: Chol-376\"'\n",
      " '\"patient id: Chol-379\"' '\"patient id: Chol-410\"'\n",
      " '\"patient id: Chol-442\"' '\"patient id: Chol-460\"'\n",
      " '\"patient id: Chol-611\"' '\"patient id: Chol-ALK-28\"'\n",
      " '\"patient id: CRC-256\"' '\"patient id: CRC-258\"' '\"patient id: CRC-272\"'\n",
      " '\"patient id: CRC-276\"' '\"patient id: CRC-304\"' '\"patient id: CRC-329\"'\n",
      " '\"patient id: CRC-342\"' '\"patient id: CRC-351\"' '\"patient id: CRC-354\"'\n",
      " '\"patient id: CRC-356\"' '\"patient id: CRC-357\"' '\"patient id: CRC-374\"'\n",
      " '\"patient id: CRC-378\"' '\"patient id: CRC-420\"' '\"patient id: CRC-429\"'\n",
      " '\"patient id: CRC-450\"' '\"patient id: CRC-453\"' '\"patient id: CRC-459\"'\n",
      " '\"patient id: CRC-462\"' '\"patient id: CRC-463\"' '\"patient id: CRC-466\"'\n",
      " '\"patient id: CRC-474\"' '\"patient id: CRC-487\"' '\"patient id: CRC-496\"'\n",
      " '\"patient id: CRC-497\"' '\"patient id: CRC-504\"' '\"patient id: CRC-511\"'\n",
      " '\"patient id: CRC-523\"' '\"patient id: CRC-524\"' '\"patient id: CRC-527\"'\n",
      " '\"patient id: CRC-531\"' '\"patient id: CRC-543\"' '\"patient id: CRC-546\"'\n",
      " '\"patient id: CRC-548\"' '\"patient id: CRC-552\"' '\"patient id: CRC-556\"'\n",
      " '\"patient id: CRC-565\"' '\"patient id: CRC-576\"' '\"patient id: GBM-280\"'\n",
      " '\"patient id: GBM-284\"' '\"patient id: GBM-306\"' '\"patient id: GBM-360\"'\n",
      " '\"patient id: GBM-372\"' '\"patient id: GBM-378\"' '\"patient id: GBM-383\"'\n",
      " '\"patient id: GBM-393\"' '\"patient id: GBM-394\"' '\"patient id: GBM-398\"'\n",
      " '\"patient id: GBM-402\"' '\"patient id: GBM-406\"' '\"patient id: GBM-408\"'\n",
      " '\"patient id: GBM-417\"' '\"patient id: GBM-418\"' '\"patient id: GBM-419\"'\n",
      " '\"patient id: GBM-421\"' '\"patient id: GBM-422\"' '\"patient id: GBM-425\"'\n",
      " '\"patient id: GBM-429\"' '\"patient id: GBM-430\"' '\"patient id: GBM-431\"'\n",
      " '\"patient id: GBM-436\"' '\"patient id: GBM-438\"' '\"patient id: GBM-439\"'\n",
      " '\"patient id: GBM-440\"' '\"patient id: GBM-443\"' '\"patient id: GBM-445\"'\n",
      " '\"patient id: GBM-454\"' '\"patient id: GBM-456\"' '\"patient id: GBM-468\"'\n",
      " '\"patient id: GBM-471\"' '\"patient id: GBM-475\"' '\"patient id: GBM-476\"'\n",
      " '\"patient id: GBM-480\"' '\"patient id: GBM-484\"' '\"patient id: GBM-485\"'\n",
      " '\"patient id: GBM-486\"' '\"patient id: GBM-499\"' '\"patient id: GBM-505\"'\n",
      " '\"patient id: HC-01\"' '\"patient id: HC-02\"' '\"patient id: HC-03\"'\n",
      " '\"patient id: HC-04\"' '\"patient id: HC-05\"' '\"patient id: HC-06\"'\n",
      " '\"patient id: HC-07\"' '\"patient id: HC-08\"' '\"patient id: HC-09\"'\n",
      " '\"patient id: HC-10\"' '\"patient id: HC-11\"' '\"patient id: HC-12\"'\n",
      " '\"patient id: HC-13\"' '\"patient id: HC-14\"' '\"patient id: HC-15\"'\n",
      " '\"patient id: HC-16\"' '\"patient id: HC-17\"' '\"patient id: HC-18\"'\n",
      " '\"patient id: HC-19\"' '\"patient id: HC-20\"' '\"patient id: HC-21\"'\n",
      " '\"patient id: HC-22\"' '\"patient id: HC-23\"' '\"patient id: HC-24\"'\n",
      " '\"patient id: HC-25\"' '\"patient id: HC-26\"' '\"patient id: HC-27\"'\n",
      " '\"patient id: HC-28\"' '\"patient id: HC-29\"' '\"patient id: HC-30\"'\n",
      " '\"patient id: HC-31\"' '\"patient id: HC-32\"' '\"patient id: HC-33\"'\n",
      " '\"patient id: HC-34\"' '\"patient id: HC-35\"' '\"patient id: HC-36\"'\n",
      " '\"patient id: HC-37\"' '\"patient id: HC-38\"' '\"patient id: HC-39\"'\n",
      " '\"patient id: HC-40\"' '\"patient id: HC-41\"' '\"patient id: HC-42\"'\n",
      " '\"patient id: HC-43\"' '\"patient id: HC-44\"' '\"patient id: HC-45\"'\n",
      " '\"patient id: HC-46\"' '\"patient id: HC-47\"' '\"patient id: HC-48\"'\n",
      " '\"patient id: HC-49\"' '\"patient id: HC-50\"' '\"patient id: HC-51\"'\n",
      " '\"patient id: HC-52\"' '\"patient id: HC-53\"' '\"patient id: HC-54\"'\n",
      " '\"patient id: HC-55\"' '\"patient id: Liver-274\"' '\"patient id: Liver-297\"'\n",
      " '\"patient id: Liver-366\"' '\"patient id: Lung-0082\"'\n",
      " '\"patient id: Lung-0096\"' '\"patient id: Lung-01\"'\n",
      " '\"patient id: Lung-0100\"' '\"patient id: Lung-0101\"'\n",
      " '\"patient id: Lung-0102\"' '\"patient id: Lung-0104\"'\n",
      " '\"patient id: Lung-0110\"' '\"patient id: Lung-0112\"'\n",
      " '\"patient id: Lung-0119\"' '\"patient id: Lung-04\"' '\"patient id: Lung-06\"'\n",
      " '\"patient id: Lung-07\"' '\"patient id: Lung-08\"' '\"patient id: Lung-12\"'\n",
      " '\"patient id: Lung-14\"' '\"patient id: Lung-15\"' '\"patient id: Lung-16\"'\n",
      " '\"patient id: Lung-17\"' '\"patient id: Lung-20\"' '\"patient id: Lung-20-1\"'\n",
      " '\"patient id: Lung-24-1\"' '\"patient id: Lung-25\"' '\"patient id: Lung-26\"'\n",
      " '\"patient id: Lung-271\"' '\"patient id: Lung-29\"' '\"patient id: Lung-30\"'\n",
      " '\"patient id: Lung-37\"' '\"patient id: Lung-40\"' '\"patient id: Lung-42\"'\n",
      " '\"patient id: Lung-46\"' '\"patient id: Lung-58\"' '\"patient id: Lung-61\"'\n",
      " '\"patient id: Lung-62\"' '\"patient id: Lung-74\"' '\"patient id: Lung-L.13\"'\n",
      " '\"patient id: Lung-L.15\"' '\"patient id: Lung-L.26\"'\n",
      " '\"patient id: Lung-L.33\"' '\"patient id: Lung-L.4\"'\n",
      " '\"patient id: Panc-260\"' '\"patient id: Panc-286\"'\n",
      " '\"patient id: Panc-314\"' '\"patient id: Panc-327\"'\n",
      " '\"patient id: Panc-328\"' '\"patient id: Panc-331\"'\n",
      " '\"patient id: Panc-335\"' '\"patient id: Panc-367\"'\n",
      " '\"patient id: Panc-369\"' '\"patient id: Panc-370\"'\n",
      " '\"patient id: Panc-371\"' '\"patient id: Panc-380\"'\n",
      " '\"patient id: Panc-382\"' '\"patient id: Panc-387\"'\n",
      " '\"patient id: Panc-388\"' '\"patient id: Panc-389\"'\n",
      " '\"patient id: Panc-390\"' '\"patient id: Panc-391\"'\n",
      " '\"patient id: Panc-431\"' '\"patient id: Panc-434\"'\n",
      " '\"patient id: Panc-435\"' '\"patient id: Panc-448\"'\n",
      " '\"patient id: Panc-449\"' '\"patient id: Panc-451\"'\n",
      " '\"patient id: Panc-461\"' '\"patient id: Panc-477\"'\n",
      " '\"patient id: Panc-488\"' '\"patient id: Panc-493\"'\n",
      " '\"patient id: Panc-498\"' '\"patient id: Panc-501\"'\n",
      " '\"patient id: Panc-503\"' '\"patient id: Panc-512\"'\n",
      " '\"patient id: Panc-553\"' '\"patient id: Panc-593\"'\n",
      " '\"patient id: Panc-597\"' '\"batch: Batch05\"' '\"batch: Batch06\"']\n"
     ]
    }
   ],
   "source": [
    "unique_cancer_types = series_df['!Sample_characteristics_ch1.2'].unique()\n",
    "print(unique_cancer_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "b9b88242-7ef1-4992-abb3-afa83d4e841d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Boolean mask for rows where column contains substring\n",
    "\n",
    "# filtered_df =  series_df[ series_df['!Sample_characteristics_ch1.3'].str.contains(\"batch:\", case=False, na=False)]\n",
    "# pd.set_option('display.max_columns', None)\n",
    "# pd.set_option('display.max_rows', None)\n",
    "# filtered_df.head()\n",
    "mask = series_df['!Sample_characteristics_ch1.2'].str.contains(\"batch\", case=False, na=False)\n",
    "\n",
    "# Swap values only for those rows\n",
    "series_df.loc[mask, ['!Sample_characteristics_ch1.2', '!Sample_characteristics_ch1.4']] = \\\n",
    "    series_df.loc[mask, ['!Sample_characteristics_ch1.4', '!Sample_characteristics_ch1.2']].values\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0034de18-c47e-425e-8e27-3c092292da64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8ce27dbb-54fe-4b37-b50f-b63bf7b241a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>\"GSM1817093\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H-11\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H11\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch05\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: PIK3CA\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817093\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>\"GSM1817094\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H31-TR-489\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H31\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817094\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>\"GSM1817095\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H38-TR467\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H38\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817095\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>\"GSM1817096\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H-59\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H59\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch05\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817096\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>\"GSM1817097\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Jul 10 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"MGH-BrCa-H63-TR-491\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"patient id: Breast-H63\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch06\"</td>\n",
       "      <td>\"cell type: thrombocytes\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1817097\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    !Sample_geo_accession           !Sample_status !Sample_submission_date  \\\n",
       "245          \"GSM1817093\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "246          \"GSM1817094\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "247          \"GSM1817095\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "248          \"GSM1817096\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "249          \"GSM1817097\"  \"Public on Oct 30 2015\"           \"Jul 10 2015\"   \n",
       "\n",
       "    !Sample_last_update_date !Sample_type !Sample_channel_count  \\\n",
       "245            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "246            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "247            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "248            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "249            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "\n",
       "    !Sample_source_name_ch1 !Sample_organism_ch1 !Sample_characteristics_ch1  \\\n",
       "245         \"MGH-BrCa-H-11\"       \"Homo sapiens\"    \"patient id: Breast-H11\"   \n",
       "246   \"MGH-BrCa-H31-TR-489\"       \"Homo sapiens\"    \"patient id: Breast-H31\"   \n",
       "247    \"MGH-BrCa-H38-TR467\"       \"Homo sapiens\"    \"patient id: Breast-H38\"   \n",
       "248         \"MGH-BrCa-H-59\"       \"Homo sapiens\"    \"patient id: Breast-H59\"   \n",
       "249   \"MGH-BrCa-H63-TR-491\"       \"Homo sapiens\"    \"patient id: Breast-H63\"   \n",
       "\n",
       "    !Sample_characteristics_ch1.1 !Sample_characteristics_ch1.2  \\\n",
       "245         \"cancer type: Breast\"              \"batch: Batch05\"   \n",
       "246         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "247         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "248         \"cancer type: Breast\"              \"batch: Batch05\"   \n",
       "249         \"cancer type: Breast\"              \"batch: Batch06\"   \n",
       "\n",
       "    !Sample_characteristics_ch1.3 !Sample_characteristics_ch1.4  \\\n",
       "245     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "246     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "247     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "248     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "249     \"cell type: thrombocytes\"               \"tissue: blood\"   \n",
       "\n",
       "     !Sample_characteristics_ch1.5 !Sample_molecule_ch1  \\\n",
       "245  \"mutational subclass: PIK3CA\"          \"total RNA\"   \n",
       "246      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "247      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "248      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "249      \"mutational subclass: wt\"          \"total RNA\"   \n",
       "\n",
       "                          !Sample_extract_protocol_ch1  \\\n",
       "245  \"Blood platelets were isolated from whole bloo...   \n",
       "246  \"Blood platelets were isolated from whole bloo...   \n",
       "247  \"Blood platelets were isolated from whole bloo...   \n",
       "248  \"Blood platelets were isolated from whole bloo...   \n",
       "249  \"Blood platelets were isolated from whole bloo...   \n",
       "\n",
       "                        !Sample_extract_protocol_ch1.1 !Sample_taxid_ch1  \\\n",
       "245  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "246  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "247  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "248  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "249  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "\n",
       "    !Sample_description                            !Sample_data_processing  \\\n",
       "245                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "246                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "247                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "248                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "249                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "\n",
       "                             !Sample_data_processing.1  \\\n",
       "245  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "246  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "247  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "248  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "249  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "\n",
       "                             !Sample_data_processing.2  \\\n",
       "245  \"Selection of intron-spanning reads was perfor...   \n",
       "246  \"Selection of intron-spanning reads was perfor...   \n",
       "247  \"Selection of intron-spanning reads was perfor...   \n",
       "248  \"Selection of intron-spanning reads was perfor...   \n",
       "249  \"Selection of intron-spanning reads was perfor...   \n",
       "\n",
       "                             !Sample_data_processing.3  \\\n",
       "245  \"Read summarization was performed using HTSeq ...   \n",
       "246  \"Read summarization was performed using HTSeq ...   \n",
       "247  \"Read summarization was performed using HTSeq ...   \n",
       "248  \"Read summarization was performed using HTSeq ...   \n",
       "249  \"Read summarization was performed using HTSeq ...   \n",
       "\n",
       "    !Sample_data_processing.4  \\\n",
       "245      \"Genome_build: hg19\"   \n",
       "246      \"Genome_build: hg19\"   \n",
       "247      \"Genome_build: hg19\"   \n",
       "248      \"Genome_build: hg19\"   \n",
       "249      \"Genome_build: hg19\"   \n",
       "\n",
       "                             !Sample_data_processing.5 !Sample_platform_id  \\\n",
       "245  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "246  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "247  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "248  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "249  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "\n",
       "    !Sample_contact_name !Sample_contact_email  \\\n",
       "245        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "246        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "247        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "248        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "249        \"Myron,,Best\"      \"m.best@vumc.nl\"   \n",
       "\n",
       "          !Sample_contact_laboratory !Sample_contact_department  \\\n",
       "245  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "246  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "247  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "248  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "249  \"Neuro-oncology Research Group\"             \"Neurosurgery\"   \n",
       "\n",
       "          !Sample_contact_institute !Sample_contact_address  \\\n",
       "245  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "246  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "247  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "248  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "249  \"VU University Medical Center\"     \"De Boelelaan 1117\"   \n",
       "\n",
       "    !Sample_contact_city !Sample_contact_zip/postal_code  \\\n",
       "245          \"Amsterdam\"                       \"1081 HV\"   \n",
       "246          \"Amsterdam\"                       \"1081 HV\"   \n",
       "247          \"Amsterdam\"                       \"1081 HV\"   \n",
       "248          \"Amsterdam\"                       \"1081 HV\"   \n",
       "249          \"Amsterdam\"                       \"1081 HV\"   \n",
       "\n",
       "    !Sample_contact_country !Sample_data_row_count !Sample_instrument_model  \\\n",
       "245           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "246           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "247           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "248           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "249           \"Netherlands\"                    \"0\"    \"Illumina HiSeq 2500\"   \n",
       "\n",
       "    !Sample_library_selection !Sample_library_source !Sample_library_strategy  \\\n",
       "245                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "246                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "247                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "248                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "249                    \"cDNA\"       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "\n",
       "                                      !Sample_relation  \\\n",
       "245  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "246  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "247  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "248  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "249  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "\n",
       "                                    !Sample_relation.1  \\\n",
       "245  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "246  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "247  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "248  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "249  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "\n",
       "    !Sample_supplementary_file_1  !series_matrix_table_begin      \"ID_REF\"  \\\n",
       "245                       \"NONE\"                         NaN  \"GSM1817093\"   \n",
       "246                       \"NONE\"                         NaN  \"GSM1817094\"   \n",
       "247                       \"NONE\"                         NaN  \"GSM1817095\"   \n",
       "248                       \"NONE\"                         NaN  \"GSM1817096\"   \n",
       "249                       \"NONE\"                         NaN  \"GSM1817097\"   \n",
       "\n",
       "     !series_matrix_table_end  \n",
       "245                       NaN  \n",
       "246                       NaN  \n",
       "247                       NaN  \n",
       "248                       NaN  \n",
       "249                       NaN  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fixing 1.1 (with 1.3)\n",
    "\n",
    "filtered_df =  series_df[ series_df['!Sample_characteristics_ch1.1'].str.contains(\"cancer\", case=False, na=False)]\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "filtered_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a661d2e5-343d-4d32-aad4-f01a6966b1da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Boolean mask for rows where column contains substring\n",
    "mask = series_df['!Sample_characteristics_ch1.1'].str.contains(\"cancer\", case=False, na=False)\n",
    "\n",
    "# Swap values only for those rows\n",
    "series_df.loc[mask, ['!Sample_characteristics_ch1.1', '!Sample_characteristics_ch1.3']] = \\\n",
    "    series_df.loc[mask, ['!Sample_characteristics_ch1.3', '!Sample_characteristics_ch1.1']].values\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a24c7d7c-92b1-4410-918f-beb4ea328c0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Boolean mask for rows where column contains substring\n",
    "mask = series_df['!Sample_characteristics_ch1'].str.contains(\"patient\", case=False, na=False)\n",
    "\n",
    "# Swap values only for those rows\n",
    "series_df.loc[mask, ['!Sample_characteristics_ch1', '!Sample_characteristics_ch1.4']] = \\\n",
    "    series_df.loc[mask, ['!Sample_characteristics_ch1.4', '!Sample_characteristics_ch1']].values\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "15f2973d-018a-40f0-b594-8968aa547403",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"mutational subclass: HER2+\"' '\"mutational subclass: wt\"'\n",
      " '\"mutational subclass: Triple Negative\"'\n",
      " '\"mutational subclass: HER2+, PIK3CA\"' '\"mutational subclass: PIK3CA\"'\n",
      " '\"mutational subclass: KRAS\"' '\"mutational subclass: MET\"'\n",
      " '\"mutational subclass: EGFR, MET\"' '\"mutational subclass: EGFR\"'\n",
      " '\"mutational subclass: KRAS, MET\"']\n"
     ]
    }
   ],
   "source": [
    "unique_mutation_subclass = series_df['!Sample_characteristics_ch1.5'].unique()\n",
    "print(unique_mutation_subclass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "6b3e5410-e4e8-41fe-9cc7-e629df1c9e73",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"patient id: Breast-03\"' '\"patient id: Breast-08\"'\n",
      " '\"patient id: Breast-10\"' '\"patient id: Breast-100\"'\n",
      " '\"patient id: Breast-15\"' '\"patient id: Breast-16\"'\n",
      " '\"patient id: Breast-21\"' '\"patient id: Breast-33\"'\n",
      " '\"patient id: Breast-42\"' '\"patient id: Breast-454\"'\n",
      " '\"patient id: Breast-48\"' '\"patient id: Breast-50\"'\n",
      " '\"patient id: Breast-55\"' '\"patient id: Breast-56\"'\n",
      " '\"patient id: Breast-564\"' '\"patient id: Breast-59\"'\n",
      " '\"patient id: Breast-590\"' '\"patient id: Breast-60\"'\n",
      " '\"patient id: Breast-603\"' '\"patient id: Breast-61\"'\n",
      " '\"patient id: Breast-85\"' '\"patient id: Breast-86\"'\n",
      " '\"patient id: Breast-97\"' '\"patient id: Breast-ALK-82\"'\n",
      " '\"patient id: Chol-292\"' '\"patient id: Chol-316\"'\n",
      " '\"patient id: Chol-341\"' '\"patient id: Chol-376\"'\n",
      " '\"patient id: Chol-379\"' '\"patient id: Chol-410\"'\n",
      " '\"patient id: Chol-442\"' '\"patient id: Chol-460\"'\n",
      " '\"patient id: Chol-611\"' '\"patient id: Chol-ALK-28\"'\n",
      " '\"patient id: CRC-256\"' '\"patient id: CRC-258\"' '\"patient id: CRC-272\"'\n",
      " '\"patient id: CRC-276\"' '\"patient id: CRC-304\"' '\"patient id: CRC-329\"'\n",
      " '\"patient id: CRC-342\"' '\"patient id: CRC-351\"' '\"patient id: CRC-354\"'\n",
      " '\"patient id: CRC-356\"' '\"patient id: CRC-357\"' '\"patient id: CRC-374\"'\n",
      " '\"patient id: CRC-378\"' '\"patient id: CRC-420\"' '\"patient id: CRC-429\"'\n",
      " '\"patient id: CRC-450\"' '\"patient id: CRC-453\"' '\"patient id: CRC-459\"'\n",
      " '\"patient id: CRC-462\"' '\"patient id: CRC-463\"' '\"patient id: CRC-466\"'\n",
      " '\"patient id: CRC-474\"' '\"patient id: CRC-487\"' '\"patient id: CRC-496\"'\n",
      " '\"patient id: CRC-497\"' '\"patient id: CRC-504\"' '\"patient id: CRC-511\"'\n",
      " '\"patient id: CRC-523\"' '\"patient id: CRC-524\"' '\"patient id: CRC-527\"'\n",
      " '\"patient id: CRC-531\"' '\"patient id: CRC-543\"' '\"patient id: CRC-546\"'\n",
      " '\"patient id: CRC-548\"' '\"patient id: CRC-552\"' '\"patient id: CRC-556\"'\n",
      " '\"patient id: CRC-565\"' '\"patient id: CRC-576\"' '\"patient id: GBM-280\"'\n",
      " '\"patient id: GBM-284\"' '\"patient id: GBM-306\"' '\"patient id: GBM-360\"'\n",
      " '\"patient id: GBM-372\"' '\"patient id: GBM-378\"' '\"patient id: GBM-383\"'\n",
      " '\"patient id: GBM-393\"' '\"patient id: GBM-394\"' '\"patient id: GBM-398\"'\n",
      " '\"patient id: GBM-402\"' '\"patient id: GBM-406\"' '\"patient id: GBM-408\"'\n",
      " '\"patient id: GBM-417\"' '\"patient id: GBM-418\"' '\"patient id: GBM-419\"'\n",
      " '\"patient id: GBM-421\"' '\"patient id: GBM-422\"' '\"patient id: GBM-425\"'\n",
      " '\"patient id: GBM-429\"' '\"patient id: GBM-430\"' '\"patient id: GBM-431\"'\n",
      " '\"patient id: GBM-436\"' '\"patient id: GBM-438\"' '\"patient id: GBM-439\"'\n",
      " '\"patient id: GBM-440\"' '\"patient id: GBM-443\"' '\"patient id: GBM-445\"'\n",
      " '\"patient id: GBM-454\"' '\"patient id: GBM-456\"' '\"patient id: GBM-468\"'\n",
      " '\"patient id: GBM-471\"' '\"patient id: GBM-475\"' '\"patient id: GBM-476\"'\n",
      " '\"patient id: GBM-480\"' '\"patient id: GBM-484\"' '\"patient id: GBM-485\"'\n",
      " '\"patient id: GBM-486\"' '\"patient id: GBM-499\"' '\"patient id: GBM-505\"'\n",
      " '\"patient id: HC-01\"' '\"patient id: HC-02\"' '\"patient id: HC-03\"'\n",
      " '\"patient id: HC-04\"' '\"patient id: HC-05\"' '\"patient id: HC-06\"'\n",
      " '\"patient id: HC-07\"' '\"patient id: HC-08\"' '\"patient id: HC-09\"'\n",
      " '\"patient id: HC-10\"' '\"patient id: HC-11\"' '\"patient id: HC-12\"'\n",
      " '\"patient id: HC-13\"' '\"patient id: HC-14\"' '\"patient id: HC-15\"'\n",
      " '\"patient id: HC-16\"' '\"patient id: HC-17\"' '\"patient id: HC-18\"'\n",
      " '\"patient id: HC-19\"' '\"patient id: HC-20\"' '\"patient id: HC-21\"'\n",
      " '\"patient id: HC-22\"' '\"patient id: HC-23\"' '\"patient id: HC-24\"'\n",
      " '\"patient id: HC-25\"' '\"patient id: HC-26\"' '\"patient id: HC-27\"'\n",
      " '\"patient id: HC-28\"' '\"patient id: HC-29\"' '\"patient id: HC-30\"'\n",
      " '\"patient id: HC-31\"' '\"patient id: HC-32\"' '\"patient id: HC-33\"'\n",
      " '\"patient id: HC-34\"' '\"patient id: HC-35\"' '\"patient id: HC-36\"'\n",
      " '\"patient id: HC-37\"' '\"patient id: HC-38\"' '\"patient id: HC-39\"'\n",
      " '\"patient id: HC-40\"' '\"patient id: HC-41\"' '\"patient id: HC-42\"'\n",
      " '\"patient id: HC-43\"' '\"patient id: HC-44\"' '\"patient id: HC-45\"'\n",
      " '\"patient id: HC-46\"' '\"patient id: HC-47\"' '\"patient id: HC-48\"'\n",
      " '\"patient id: HC-49\"' '\"patient id: HC-50\"' '\"patient id: HC-51\"'\n",
      " '\"patient id: HC-52\"' '\"patient id: HC-53\"' '\"patient id: HC-54\"'\n",
      " '\"patient id: HC-55\"' '\"patient id: Liver-274\"' '\"patient id: Liver-297\"'\n",
      " '\"patient id: Liver-366\"' '\"patient id: Lung-0082\"'\n",
      " '\"patient id: Lung-0096\"' '\"patient id: Lung-01\"'\n",
      " '\"patient id: Lung-0100\"' '\"patient id: Lung-0101\"'\n",
      " '\"patient id: Lung-0102\"' '\"patient id: Lung-0104\"'\n",
      " '\"patient id: Lung-0110\"' '\"patient id: Lung-0112\"'\n",
      " '\"patient id: Lung-0119\"' '\"patient id: Lung-04\"' '\"patient id: Lung-06\"'\n",
      " '\"patient id: Lung-07\"' '\"patient id: Lung-08\"' '\"patient id: Lung-12\"'\n",
      " '\"patient id: Lung-14\"' '\"patient id: Lung-15\"' '\"patient id: Lung-16\"'\n",
      " '\"patient id: Lung-17\"' '\"patient id: Lung-20\"' '\"patient id: Lung-20-1\"'\n",
      " '\"patient id: Lung-24-1\"' '\"patient id: Lung-25\"' '\"patient id: Lung-26\"'\n",
      " '\"patient id: Lung-271\"' '\"patient id: Lung-29\"' '\"patient id: Lung-30\"'\n",
      " '\"patient id: Lung-37\"' '\"patient id: Lung-40\"' '\"patient id: Lung-42\"'\n",
      " '\"patient id: Lung-46\"' '\"patient id: Lung-58\"' '\"patient id: Lung-61\"'\n",
      " '\"patient id: Lung-62\"' '\"patient id: Lung-74\"' '\"patient id: Lung-L.13\"'\n",
      " '\"patient id: Lung-L.15\"' '\"patient id: Lung-L.26\"'\n",
      " '\"patient id: Lung-L.33\"' '\"patient id: Lung-L.4\"'\n",
      " '\"patient id: Panc-260\"' '\"patient id: Panc-286\"'\n",
      " '\"patient id: Panc-314\"' '\"patient id: Panc-327\"'\n",
      " '\"patient id: Panc-328\"' '\"patient id: Panc-331\"'\n",
      " '\"patient id: Panc-335\"' '\"patient id: Panc-367\"'\n",
      " '\"patient id: Panc-369\"' '\"patient id: Panc-370\"'\n",
      " '\"patient id: Panc-371\"' '\"patient id: Panc-380\"'\n",
      " '\"patient id: Panc-382\"' '\"patient id: Panc-387\"'\n",
      " '\"patient id: Panc-388\"' '\"patient id: Panc-389\"'\n",
      " '\"patient id: Panc-390\"' '\"patient id: Panc-391\"'\n",
      " '\"patient id: Panc-431\"' '\"patient id: Panc-434\"'\n",
      " '\"patient id: Panc-435\"' '\"patient id: Panc-448\"'\n",
      " '\"patient id: Panc-449\"' '\"patient id: Panc-451\"'\n",
      " '\"patient id: Panc-461\"' '\"patient id: Panc-477\"'\n",
      " '\"patient id: Panc-488\"' '\"patient id: Panc-493\"'\n",
      " '\"patient id: Panc-498\"' '\"patient id: Panc-501\"'\n",
      " '\"patient id: Panc-503\"' '\"patient id: Panc-512\"'\n",
      " '\"patient id: Panc-553\"' '\"patient id: Panc-593\"'\n",
      " '\"patient id: Panc-597\"' '\"batch: Batch05\"' '\"batch: Batch06\"']\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2ce4daf2-1d85-4e2a-bba3-b9e51bcbbd96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"batch: Batch03\"' '\"batch: Batch04\"' '\"batch: Batch02\"'\n",
      " '\"batch: Batch01\"' '\"patient id: Breast-H11\"' '\"patient id: Breast-H31\"'\n",
      " '\"patient id: Breast-H38\"' '\"patient id: Breast-H59\"'\n",
      " '\"patient id: Breast-H63\"' '\"patient id: Breast-H66\"'\n",
      " '\"patient id: Breast-H68\"' '\"patient id: Breast-H74\"'\n",
      " '\"patient id: Breast-H75\"' '\"patient id: Breast-H76\"'\n",
      " '\"patient id: Breast-H81\"' '\"patient id: Breast-H83\"'\n",
      " '\"patient id: Breast-H84\"' '\"patient id: Breast-H89\"'\n",
      " '\"patient id: Breast-H92\"' '\"patient id: Chol-BRAF4\"'\n",
      " '\"patient id: CRC-368\"' '\"patient id: CRC-412\"' '\"patient id: CRC-BRAF5\"'\n",
      " '\"patient id: CRC-BRAF6\"' '\"patient id: Lung-49\"' '\"patient id: Lung-57\"'\n",
      " '\"patient id: Lung-59\"' '\"patient id: Lung-L01\"' '\"patient id: Lung-L03\"'\n",
      " '\"patient id: Lung-L07\"' '\"patient id: Lung-L11\"'\n",
      " '\"patient id: Lung-L12\"' '\"patient id: Lung-L18\"'\n",
      " '\"patient id: Lung-L19\"' '\"patient id: Lung-L20\"'\n",
      " '\"patient id: Lung-L22\"' '\"patient id: Lung-L23\"'\n",
      " '\"patient id: Lung-L25\"' '\"patient id: Lung-L39\"'\n",
      " '\"patient id: Lung-L40\"' '\"patient id: Lung-L51\"'\n",
      " '\"patient id: Lung-L58\"' '\"patient id: Lung-L59\"'\n",
      " '\"patient id: Lung-L65\"']\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d6a2d013-1f39-47df-b486-fe67901c2494",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"cell type: Thrombocytes\"' '\"cell type: thrombocytes\"']\n"
     ]
    }
   ],
   "source": [
    "unique_cell_type = series_df['!Sample_characteristics_ch1.1'].unique()\n",
    "print(unique_cell_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "b892d01c-d3c4-46f9-9303-0f8423ab2aeb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"tissue: blood\"']\n"
     ]
    }
   ],
   "source": [
    "unique_cancer_types = series_df['!Sample_characteristics_ch1'].unique()\n",
    "print(unique_cancer_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1535d056-9e08-4bd6-a074-3a538b3a1979",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"cancer type: Breast\"' '\"cancer type: Hepatobiliary\"'\n",
      " '\"cancer type: CRC\"' '\"cancer type: GBM\"' '\"cancer type: HC\"'\n",
      " '\"cancer type: Lung\"' '\"cancer type: Pancreas\"']\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ace0fff2-aa87-48cc-99eb-ee0fa57636e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"cancer type: Breast\"' '\"cancer type: Hepatobiliary\"'\n",
      " '\"cancer type: CRC\"' '\"cancer type: GBM\"' '\"cancer type: HC\"'\n",
      " '\"cancer type: Lung\"' '\"cancer type: Pancreas\"']\n"
     ]
    }
   ],
   "source": [
    "unique_cancer_types = series_df['!Sample_characteristics_ch1.3'].unique()\n",
    "print(unique_cancer_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "466e57bb-1804-4bb7-a615-9f1ade667320",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "8be7be28-7f33-4485-950f-c3deefc59c55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [!Sample_geo_accession, !Sample_status, !Sample_submission_date, !Sample_last_update_date, !Sample_type, !Sample_channel_count, !Sample_source_name_ch1, !Sample_organism_ch1, !Sample_characteristics_ch1, !Sample_characteristics_ch1.1, !Sample_characteristics_ch1.2, !Sample_characteristics_ch1.3, !Sample_characteristics_ch1.4, !Sample_characteristics_ch1.5, !Sample_molecule_ch1, !Sample_extract_protocol_ch1, !Sample_extract_protocol_ch1.1, !Sample_taxid_ch1, !Sample_description, !Sample_data_processing, !Sample_data_processing.1, !Sample_data_processing.2, !Sample_data_processing.3, !Sample_data_processing.4, !Sample_data_processing.5, !Sample_platform_id, !Sample_contact_name, !Sample_contact_email, !Sample_contact_laboratory, !Sample_contact_department, !Sample_contact_institute, !Sample_contact_address, !Sample_contact_city, !Sample_contact_zip/postal_code, !Sample_contact_country, !Sample_data_row_count, !Sample_instrument_model, !Sample_library_selection, !Sample_library_source, !Sample_library_strategy, !Sample_relation, !Sample_relation.1, !Sample_supplementary_file_1, !series_matrix_table_begin, \"ID_REF\", !series_matrix_table_end]\n",
       "Index: []"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fixing 1.3\n",
    "\n",
    "filtered_df =  series_df[ series_df['!Sample_characteristics_ch1'].str.contains(\"mutational subclass:\", case=False, na=False)]\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "filtered_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc0fb217-f4f8-4f6b-9030-9bcbff5a1367",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ea53af8c-4853-4afe-aa7c-9a049f73e334",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unnamed: 0                   0\n",
      "3-Breast-Her2-ampl           0\n",
      "8-Breast-WT                  0\n",
      "10-Breast-Her2-ampl          0\n",
      "Breast-100                   0\n",
      "15-Breast-Her2-ampl          0\n",
      "16-Breast-WT                 0\n",
      "21-Breast-WT                 0\n",
      "33-Breast-Her2-ampl          0\n",
      "42-Breast-Her2-ampl          0\n",
      "Breast-454                   0\n",
      "48-Breast-WT                 0\n",
      "50-Breast-WT                 0\n",
      "55-Breast-WT                 0\n",
      "Type-Unknown-6               0\n",
      "Breast-564                   0\n",
      "59-Breast-WT                 0\n",
      "Breast-590                   0\n",
      "Breast-60                    0\n",
      "Breast-603                   0\n",
      "ALK61-Breast-Her2-unknown    0\n",
      "Breast-85                    0\n",
      "Breast-86                    0\n",
      "97-Breast-Her2-ampl          0\n",
      "Breast-ALK-82                0\n",
      "292-Liver-KRAS               0\n",
      "316-Liver-KRAS               0\n",
      "341-Liver-WT                 0\n",
      "376-Liver-WT                 0\n",
      "379-Liver-WT                 0\n",
      "Chol-410                     0\n",
      "Chol-442                     0\n",
      "Chol-460                     0\n",
      "Chol-611                     0\n",
      "Chol-ALK-28                  0\n",
      "VU256-CRC                    0\n",
      "VU258-CRC                    0\n",
      "VU272-CRC                    0\n",
      "VU276-CRC                    0\n",
      "Type-Unknown-1               0\n",
      "329-CRC-KRAS                 0\n",
      "342-CRC-KRAS                 0\n",
      "Type-Unknown-5               0\n",
      "354-CRC-WT                   0\n",
      "356-CRC-WT                   0\n",
      "CRC-357                      0\n",
      "374-CRC-WT                   0\n",
      "378-CRC-WT                   0\n",
      "420-CRC-WT                   0\n",
      "429-CRC-WT                   0\n",
      "CRC-450                      0\n",
      "453-CRC-KRAS                 0\n",
      "CRC-459                      0\n",
      "462-CRC-KRAS                 0\n",
      "CRC-463                      0\n",
      "466-CRC-KRAS                 0\n",
      "474-CRC-WT                   0\n",
      "CRC-487                      0\n",
      "CRC-496                      0\n",
      "497-CRC-KRAS                 0\n",
      "504-CRC-WT                   0\n",
      "CRC-511                      0\n",
      "CRC-523                      0\n",
      "CRC-524                      0\n",
      "527-CRC-WT                   0\n",
      "CRC-531                      0\n",
      "543-CRC-KRAS                 0\n",
      "546-CRC-KRAS                 0\n",
      "548-CRC-KRAS                 0\n",
      "CRC-552                      0\n",
      "556-CRC-WT                   0\n",
      "CRC-565                      0\n",
      "CRC-576                      0\n",
      "VU280-GBM                    0\n",
      "VU284-GBM-vIII               0\n",
      "VU306-GBM-vIII               0\n",
      "VU360-GBM-WT                 0\n",
      "VU372-GBM-vIII               0\n",
      "VU378-GBM                    0\n",
      "VU383Platelet-hiseq          0\n",
      "VU393-GBM-vIII               0\n",
      "VU394Platelet-hiseq          0\n",
      "VU398-GBM                    0\n",
      "VU402-2-GBM                  0\n",
      "VU406-2-GBM                  0\n",
      "VU408-GBM                    0\n",
      "VU417-GBM-vIII               0\n",
      "VU418-GBM                    0\n",
      "VU419-GBM                    0\n",
      "VU421-GBM                    0\n",
      "VU422-GBM                    0\n",
      "VU425-GBM                    0\n",
      "VU429-1-GBM-vIII             0\n",
      "VU430-GBM                    0\n",
      "VU431-1-GBM-vIII             0\n",
      "VU436-1-GBM-vIII             0\n",
      "GBM-438                      0\n",
      "VU439-1-GBM-vIII             0\n",
      "GBM-440                      0\n",
      "VU443-1-GBM-WT               0\n",
      "VU445-1-GBM-WT               0\n",
      "VU454-1-GBM-WT               0\n",
      "VU456-1-GBM-WT               0\n",
      "VU468-1-GBM-WT               0\n",
      "GBM-471                      0\n",
      "GBM-475                      0\n",
      "GBM-476                      0\n",
      "GBM-480                      0\n",
      "GBM-484                      0\n",
      "GBM-485                      0\n",
      "GBM-486                      0\n",
      "GBM-499                      0\n",
      "GBM-505                      0\n",
      "HD-1                         0\n",
      "HD-2-1                       0\n",
      "HD-3-1                       0\n",
      "HD-4                         0\n",
      "HD-5                         0\n",
      "HD-7                         0\n",
      "HD-8                         0\n",
      "HD-9                         0\n",
      "HD-10                        0\n",
      "HD-11-1                      0\n",
      "HD-13-1                      0\n",
      "HD-14-1                      0\n",
      "HD-15-1                      0\n",
      "HD-16-1                      0\n",
      "HD-20-1                      0\n",
      "HD-22-1                      0\n",
      "HD-34-1                      0\n",
      "HD-39-1                      0\n",
      "HD-40-1                      0\n",
      "HD-45-1                      0\n",
      "HD-46-1                      0\n",
      "Control-2                    0\n",
      "Control-3                    0\n",
      "Control-4                    0\n",
      "Control-5                    0\n",
      "Control-6                    0\n",
      "Control-8                    0\n",
      "Control-9                    0\n",
      "Control-10                   0\n",
      "Control-13                   0\n",
      "Type-Unknown-3               0\n",
      "HD-17-1                      0\n",
      "HD-18-2                      0\n",
      "HD-21-1                      0\n",
      "HD-24-2                      0\n",
      "HD-25-2                      0\n",
      "HD-26-2                      0\n",
      "HD-27-2                      0\n",
      "HD-29-2                      0\n",
      "HD-30-2                      0\n",
      "HD-31                        0\n",
      "HD-32-2                      0\n",
      "HD-33-1                      0\n",
      "HD-35-2                      0\n",
      "HD-36                        0\n",
      "HD-37-2                      0\n",
      "HD-38-2                      0\n",
      "HD-44-2                      0\n",
      "HD-47-1                      0\n",
      "HD-48-1                      0\n",
      "HD-49-1                      0\n",
      "HD-50-1                      0\n",
      "HD-51-1                      0\n",
      "HD-53-1                      0\n",
      "HD-54-1                      0\n",
      "VU274-CRC                    0\n",
      "297-Liver-WT                 0\n",
      "366-Liver-WT                 0\n",
      "Lung-0082                    0\n",
      "Lung-0096                    0\n",
      "Lung-0001                    0\n",
      "Lung-0100                    0\n",
      "Lung-0101                    0\n",
      "Lung-0102                    0\n",
      "Lung-0104                    0\n",
      "Lung-0110                    0\n",
      "Lung-0112                    0\n",
      "Lung-0119                    0\n",
      "Lung-004-EGFR                0\n",
      "Lung-006                     0\n",
      "Lung-0007                    0\n",
      "Lung-008                     0\n",
      "Lung-0012                    0\n",
      "Lung-0014J                   0\n",
      "Lung-0015                    0\n",
      "Lung-0016                    0\n",
      "Lung-0017                    0\n",
      "Lung-0020J                   0\n",
      "Lung-0020J-1                 0\n",
      "Lung-0024J-1                 0\n",
      "Lung-0025J                   0\n",
      "Lung-0026J                   0\n",
      "VU271-CRC                    0\n",
      "Lung-0029J-1-WT              0\n",
      "Lung-0030J-1-WT              0\n",
      "Lung-0037                    0\n",
      "Lung-0040                    0\n",
      "Lung-0042                    0\n",
      "Lung-0046                    0\n",
      "Lung-0058-EGFR               0\n",
      "Lung-0061-EGFR               0\n",
      "Lung-0062-EGFR               0\n",
      "Lung-0074-EGFR               0\n",
      "Lung-L-13                    0\n",
      "Lung-L-15                    0\n",
      "Lung-L-26                    0\n",
      "Lung-L-33                    0\n",
      "Lung-L-4                     0\n",
      "VU260-CRC                    0\n",
      "286-Pancr-286-WT             0\n",
      "314-Pancr-WT                 0\n",
      "327-Pancr-KRAS               0\n",
      "328-Pancr-WT                 0\n",
      "331-Pancr-WT                 0\n",
      "Panc-335                     0\n",
      "367-Pancr-KRAS               0\n",
      "369-Pancr-KRAS               0\n",
      "Panc-370                     0\n",
      "371-Pancr-KRAS               0\n",
      "380-Pancr-KRAS               0\n",
      "382-Pancr-KRAS               0\n",
      "387-Pancr-WT                 0\n",
      "388-Pancr-WT                 0\n",
      "389-Pancr-WT                 0\n",
      "390-Pancr-KRAS               0\n",
      "391-Pancr-KRAS               0\n",
      "431-Pancr-WT                 0\n",
      "434-Pancr-WT                 0\n",
      "Panc-435                     0\n",
      "448-Pancr-KRAS               0\n",
      "Type-Unknown-4               0\n",
      "Panc-451                     0\n",
      "Panc-461                     0\n",
      "Panc-477                     0\n",
      "488-Pancr-WT                 0\n",
      "Panc-493                     0\n",
      "498-Pancr-WT                 0\n",
      "Panc-501                     0\n",
      "Panc-503                     0\n",
      "512-Pancr-KRAS               0\n",
      "553-Pancr-KRAS               0\n",
      "Panc-593                     0\n",
      "Panc-597                     0\n",
      "MGH-BrCa-H-11                0\n",
      "MGH-BrCa-H-59                0\n",
      "MGH-BrCa-H-66                0\n",
      "MGH-BrCa-H-68                0\n",
      "MGH-BrCa-H-74                0\n",
      "MGH-BrCa-H-75                0\n",
      "MGH-CRC-368                  0\n",
      "Vumc-NSCLC-49                0\n",
      "Vumc-NSCLC-57                0\n",
      "Vumc-NSCLC-59                0\n",
      "MGH-BrCa-H31-TR-489          0\n",
      "MGH-BrCa-H38-TR467           0\n",
      "MGH-BrCa-H63-TR-491          0\n",
      "MGH-BrCa-H76-TR469           0\n",
      "MGH-BrCa-H81-TR492           0\n",
      "MGH-BrCa-H83-TR550           0\n",
      "MGH-BrCa-H84-TR551           0\n",
      "MGH-BrCa-H89-TR471           0\n",
      "MGH-BrCa-H92-TR472           0\n",
      "MGH-CRC-412-TR466            0\n",
      "MGH-CRC-BRAF4-TR547          0\n",
      "MGH-CRC-BRAF5-TR548          0\n",
      "MGH-CRC-BRAF6-TR465          0\n",
      "MGH-NSCLC-L01-TR458          0\n",
      "MGH-NSCLC-L03-TR488          0\n",
      "MGH-NSCLC-L07-TR459          0\n",
      "MGH-NSCLC-L11-TR477          0\n",
      "MGH-NSCLC-L12-TR478          0\n",
      "MGH-NSCLC-L18-TR479          0\n",
      "MGH-NSCLC-L19-TR461          0\n",
      "MGH-NSCLC-L20-TR500          0\n",
      "MGH-NSCLC-L22-TR462          0\n",
      "MGH-NSCLC-L23-TR524          0\n",
      "MGH-NSCLC-L25-TR480          0\n",
      "MGH-NSCLC-L39-TR519          0\n",
      "MGH-NSCLC-L40-TR520          0\n",
      "MGH-NSCLC-L51-TR521          0\n",
      "MGH-NSCLC-L58-TR525          0\n",
      "MGH-NSCLC-L59-TR522          0\n",
      "MGH-NSCLC-L65-TR523          0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(tep_df.isnull().sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "6cadd074-f420-412c-baf5-9e538813c59a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"mutational subclass: HER2+\"' '\"mutational subclass: wt\"'\n",
      " '\"mutational subclass: Triple Negative\"'\n",
      " '\"mutational subclass: HER2+, PIK3CA\"' '\"mutational subclass: PIK3CA\"'\n",
      " '\"mutational subclass: KRAS\"' '\"mutational subclass: MET\"'\n",
      " '\"mutational subclass: EGFR, MET\"' '\"mutational subclass: EGFR\"'\n",
      " '\"mutational subclass: KRAS, MET\"']\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "4c4bec85-e74c-4b09-896a-6c5fbce553e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"tissue: blood\"']\n",
      "['\"patient id: Breast-03\"' '\"patient id: Breast-08\"'\n",
      " '\"patient id: Breast-10\"' '\"patient id: Breast-100\"'\n",
      " '\"patient id: Breast-15\"' '\"patient id: Breast-16\"'\n",
      " '\"patient id: Breast-21\"' '\"patient id: Breast-33\"'\n",
      " '\"patient id: Breast-42\"' '\"patient id: Breast-454\"'\n",
      " '\"patient id: Breast-48\"' '\"patient id: Breast-50\"'\n",
      " '\"patient id: Breast-55\"' '\"patient id: Breast-56\"'\n",
      " '\"patient id: Breast-564\"' '\"patient id: Breast-59\"'\n",
      " '\"patient id: Breast-590\"' '\"patient id: Breast-60\"'\n",
      " '\"patient id: Breast-603\"' '\"patient id: Breast-61\"'\n",
      " '\"patient id: Breast-85\"' '\"patient id: Breast-86\"'\n",
      " '\"patient id: Breast-97\"' '\"patient id: Breast-ALK-82\"'\n",
      " '\"patient id: Chol-292\"' '\"patient id: Chol-316\"'\n",
      " '\"patient id: Chol-341\"' '\"patient id: Chol-376\"'\n",
      " '\"patient id: Chol-379\"' '\"patient id: Chol-410\"'\n",
      " '\"patient id: Chol-442\"' '\"patient id: Chol-460\"'\n",
      " '\"patient id: Chol-611\"' '\"patient id: Chol-ALK-28\"'\n",
      " '\"patient id: CRC-256\"' '\"patient id: CRC-258\"' '\"patient id: CRC-272\"'\n",
      " '\"patient id: CRC-276\"' '\"patient id: CRC-304\"' '\"patient id: CRC-329\"'\n",
      " '\"patient id: CRC-342\"' '\"patient id: CRC-351\"' '\"patient id: CRC-354\"'\n",
      " '\"patient id: CRC-356\"' '\"patient id: CRC-357\"' '\"patient id: CRC-374\"'\n",
      " '\"patient id: CRC-378\"' '\"patient id: CRC-420\"' '\"patient id: CRC-429\"'\n",
      " '\"patient id: CRC-450\"' '\"patient id: CRC-453\"' '\"patient id: CRC-459\"'\n",
      " '\"patient id: CRC-462\"' '\"patient id: CRC-463\"' '\"patient id: CRC-466\"'\n",
      " '\"patient id: CRC-474\"' '\"patient id: CRC-487\"' '\"patient id: CRC-496\"'\n",
      " '\"patient id: CRC-497\"' '\"patient id: CRC-504\"' '\"patient id: CRC-511\"'\n",
      " '\"patient id: CRC-523\"' '\"patient id: CRC-524\"' '\"patient id: CRC-527\"'\n",
      " '\"patient id: CRC-531\"' '\"patient id: CRC-543\"' '\"patient id: CRC-546\"'\n",
      " '\"patient id: CRC-548\"' '\"patient id: CRC-552\"' '\"patient id: CRC-556\"'\n",
      " '\"patient id: CRC-565\"' '\"patient id: CRC-576\"' '\"patient id: GBM-280\"'\n",
      " '\"patient id: GBM-284\"' '\"patient id: GBM-306\"' '\"patient id: GBM-360\"'\n",
      " '\"patient id: GBM-372\"' '\"patient id: GBM-378\"' '\"patient id: GBM-383\"'\n",
      " '\"patient id: GBM-393\"' '\"patient id: GBM-394\"' '\"patient id: GBM-398\"'\n",
      " '\"patient id: GBM-402\"' '\"patient id: GBM-406\"' '\"patient id: GBM-408\"'\n",
      " '\"patient id: GBM-417\"' '\"patient id: GBM-418\"' '\"patient id: GBM-419\"'\n",
      " '\"patient id: GBM-421\"' '\"patient id: GBM-422\"' '\"patient id: GBM-425\"'\n",
      " '\"patient id: GBM-429\"' '\"patient id: GBM-430\"' '\"patient id: GBM-431\"'\n",
      " '\"patient id: GBM-436\"' '\"patient id: GBM-438\"' '\"patient id: GBM-439\"'\n",
      " '\"patient id: GBM-440\"' '\"patient id: GBM-443\"' '\"patient id: GBM-445\"'\n",
      " '\"patient id: GBM-454\"' '\"patient id: GBM-456\"' '\"patient id: GBM-468\"'\n",
      " '\"patient id: GBM-471\"' '\"patient id: GBM-475\"' '\"patient id: GBM-476\"'\n",
      " '\"patient id: GBM-480\"' '\"patient id: GBM-484\"' '\"patient id: GBM-485\"'\n",
      " '\"patient id: GBM-486\"' '\"patient id: GBM-499\"' '\"patient id: GBM-505\"'\n",
      " '\"patient id: HC-01\"' '\"patient id: HC-02\"' '\"patient id: HC-03\"'\n",
      " '\"patient id: HC-04\"' '\"patient id: HC-05\"' '\"patient id: HC-06\"'\n",
      " '\"patient id: HC-07\"' '\"patient id: HC-08\"' '\"patient id: HC-09\"'\n",
      " '\"patient id: HC-10\"' '\"patient id: HC-11\"' '\"patient id: HC-12\"'\n",
      " '\"patient id: HC-13\"' '\"patient id: HC-14\"' '\"patient id: HC-15\"'\n",
      " '\"patient id: HC-16\"' '\"patient id: HC-17\"' '\"patient id: HC-18\"'\n",
      " '\"patient id: HC-19\"' '\"patient id: HC-20\"' '\"patient id: HC-21\"'\n",
      " '\"patient id: HC-22\"' '\"patient id: HC-23\"' '\"patient id: HC-24\"'\n",
      " '\"patient id: HC-25\"' '\"patient id: HC-26\"' '\"patient id: HC-27\"'\n",
      " '\"patient id: HC-28\"' '\"patient id: HC-29\"' '\"patient id: HC-30\"'\n",
      " '\"patient id: HC-31\"' '\"patient id: HC-32\"' '\"patient id: HC-33\"'\n",
      " '\"patient id: HC-34\"' '\"patient id: HC-35\"' '\"patient id: HC-36\"'\n",
      " '\"patient id: HC-37\"' '\"patient id: HC-38\"' '\"patient id: HC-39\"'\n",
      " '\"patient id: HC-40\"' '\"patient id: HC-41\"' '\"patient id: HC-42\"'\n",
      " '\"patient id: HC-43\"' '\"patient id: HC-44\"' '\"patient id: HC-45\"'\n",
      " '\"patient id: HC-46\"' '\"patient id: HC-47\"' '\"patient id: HC-48\"'\n",
      " '\"patient id: HC-49\"' '\"patient id: HC-50\"' '\"patient id: HC-51\"'\n",
      " '\"patient id: HC-52\"' '\"patient id: HC-53\"' '\"patient id: HC-54\"'\n",
      " '\"patient id: HC-55\"' '\"patient id: Liver-274\"' '\"patient id: Liver-297\"'\n",
      " '\"patient id: Liver-366\"' '\"patient id: Lung-0082\"'\n",
      " '\"patient id: Lung-0096\"' '\"patient id: Lung-01\"'\n",
      " '\"patient id: Lung-0100\"' '\"patient id: Lung-0101\"'\n",
      " '\"patient id: Lung-0102\"' '\"patient id: Lung-0104\"'\n",
      " '\"patient id: Lung-0110\"' '\"patient id: Lung-0112\"'\n",
      " '\"patient id: Lung-0119\"' '\"patient id: Lung-04\"' '\"patient id: Lung-06\"'\n",
      " '\"patient id: Lung-07\"' '\"patient id: Lung-08\"' '\"patient id: Lung-12\"'\n",
      " '\"patient id: Lung-14\"' '\"patient id: Lung-15\"' '\"patient id: Lung-16\"'\n",
      " '\"patient id: Lung-17\"' '\"patient id: Lung-20\"' '\"patient id: Lung-20-1\"'\n",
      " '\"patient id: Lung-24-1\"' '\"patient id: Lung-25\"' '\"patient id: Lung-26\"'\n",
      " '\"patient id: Lung-271\"' '\"patient id: Lung-29\"' '\"patient id: Lung-30\"'\n",
      " '\"patient id: Lung-37\"' '\"patient id: Lung-40\"' '\"patient id: Lung-42\"'\n",
      " '\"patient id: Lung-46\"' '\"patient id: Lung-58\"' '\"patient id: Lung-61\"'\n",
      " '\"patient id: Lung-62\"' '\"patient id: Lung-74\"' '\"patient id: Lung-L.13\"'\n",
      " '\"patient id: Lung-L.15\"' '\"patient id: Lung-L.26\"'\n",
      " '\"patient id: Lung-L.33\"' '\"patient id: Lung-L.4\"'\n",
      " '\"patient id: Panc-260\"' '\"patient id: Panc-286\"'\n",
      " '\"patient id: Panc-314\"' '\"patient id: Panc-327\"'\n",
      " '\"patient id: Panc-328\"' '\"patient id: Panc-331\"'\n",
      " '\"patient id: Panc-335\"' '\"patient id: Panc-367\"'\n",
      " '\"patient id: Panc-369\"' '\"patient id: Panc-370\"'\n",
      " '\"patient id: Panc-371\"' '\"patient id: Panc-380\"'\n",
      " '\"patient id: Panc-382\"' '\"patient id: Panc-387\"'\n",
      " '\"patient id: Panc-388\"' '\"patient id: Panc-389\"'\n",
      " '\"patient id: Panc-390\"' '\"patient id: Panc-391\"'\n",
      " '\"patient id: Panc-431\"' '\"patient id: Panc-434\"'\n",
      " '\"patient id: Panc-435\"' '\"patient id: Panc-448\"'\n",
      " '\"patient id: Panc-449\"' '\"patient id: Panc-451\"'\n",
      " '\"patient id: Panc-461\"' '\"patient id: Panc-477\"'\n",
      " '\"patient id: Panc-488\"' '\"patient id: Panc-493\"'\n",
      " '\"patient id: Panc-498\"' '\"patient id: Panc-501\"'\n",
      " '\"patient id: Panc-503\"' '\"patient id: Panc-512\"'\n",
      " '\"patient id: Panc-553\"' '\"patient id: Panc-593\"'\n",
      " '\"patient id: Panc-597\"' '\"patient id: Breast-H11\"'\n",
      " '\"patient id: Breast-H31\"' '\"patient id: Breast-H38\"'\n",
      " '\"patient id: Breast-H59\"' '\"patient id: Breast-H63\"'\n",
      " '\"patient id: Breast-H66\"' '\"patient id: Breast-H68\"'\n",
      " '\"patient id: Breast-H74\"' '\"patient id: Breast-H75\"'\n",
      " '\"patient id: Breast-H76\"' '\"patient id: Breast-H81\"'\n",
      " '\"patient id: Breast-H83\"' '\"patient id: Breast-H84\"'\n",
      " '\"patient id: Breast-H89\"' '\"patient id: Breast-H92\"'\n",
      " '\"patient id: Chol-BRAF4\"' '\"patient id: CRC-368\"'\n",
      " '\"patient id: CRC-412\"' '\"patient id: CRC-BRAF5\"'\n",
      " '\"patient id: CRC-BRAF6\"' '\"patient id: Lung-49\"' '\"patient id: Lung-57\"'\n",
      " '\"patient id: Lung-59\"' '\"patient id: Lung-L01\"' '\"patient id: Lung-L03\"'\n",
      " '\"patient id: Lung-L07\"' '\"patient id: Lung-L11\"'\n",
      " '\"patient id: Lung-L12\"' '\"patient id: Lung-L18\"'\n",
      " '\"patient id: Lung-L19\"' '\"patient id: Lung-L20\"'\n",
      " '\"patient id: Lung-L22\"' '\"patient id: Lung-L23\"'\n",
      " '\"patient id: Lung-L25\"' '\"patient id: Lung-L39\"'\n",
      " '\"patient id: Lung-L40\"' '\"patient id: Lung-L51\"'\n",
      " '\"patient id: Lung-L58\"' '\"patient id: Lung-L59\"'\n",
      " '\"patient id: Lung-L65\"']\n",
      "['\"cancer type: Breast\"' '\"cancer type: Hepatobiliary\"'\n",
      " '\"cancer type: CRC\"' '\"cancer type: GBM\"' '\"cancer type: HC\"'\n",
      " '\"cancer type: Lung\"' '\"cancer type: Pancreas\"']\n",
      "['\"batch: Batch03\"' '\"batch: Batch04\"' '\"batch: Batch02\"'\n",
      " '\"batch: Batch01\"' '\"batch: Batch05\"' '\"batch: Batch06\"']\n",
      "['\"mutational subclass: HER2+\"' '\"mutational subclass: wt\"'\n",
      " '\"mutational subclass: Triple Negative\"'\n",
      " '\"mutational subclass: HER2+, PIK3CA\"' '\"mutational subclass: PIK3CA\"'\n",
      " '\"mutational subclass: KRAS\"' '\"mutational subclass: MET\"'\n",
      " '\"mutational subclass: EGFR, MET\"' '\"mutational subclass: EGFR\"'\n",
      " '\"mutational subclass: KRAS, MET\"']\n"
     ]
    }
   ],
   "source": [
    "unique_tissue = series_df['!Sample_characteristics_ch1'].unique()\n",
    "print(unique_tissue)\n",
    "\n",
    "unique_patient_id = series_df['!Sample_characteristics_ch1.2'].unique()\n",
    "print(unique_patient_id)\n",
    "\n",
    "unique_cancer_types = series_df['!Sample_characteristics_ch1.3'].unique()\n",
    "print(unique_cancer_types)\n",
    "\n",
    "unique_batch_id = series_df['!Sample_characteristics_ch1.4'].unique()\n",
    "print(unique_batch_id)\n",
    "\n",
    "unique_mutation_subclass = series_df['!Sample_characteristics_ch1.5'].unique()\n",
    "print(unique_mutation_subclass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "bee912eb-e187-4f26-aae9-5aab10d32717",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df.to_csv('GSE68086_series_matrix_fixed.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "127c031b-7849-47f6-b963-a31832753976",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
