{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4c32b6ad-0d33-48df-a22e-a3d46573e156",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "24a02fdc-549b-4fa9-b4dd-45304814caba",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df = pd.read_csv(\"GSE68086_series_matrix_fixed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b4fd97a2-863f-46db-85ad-b752888d43d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['\"tissue: blood\"']\n",
      "['\"patient id: Breast-03\"' '\"patient id: Breast-08\"'\n",
      " '\"patient id: Breast-10\"' '\"patient id: Breast-100\"'\n",
      " '\"patient id: Breast-15\"' '\"patient id: Breast-16\"'\n",
      " '\"patient id: Breast-21\"' '\"patient id: Breast-33\"'\n",
      " '\"patient id: Breast-42\"' '\"patient id: Breast-454\"'\n",
      " '\"patient id: Breast-48\"' '\"patient id: Breast-50\"'\n",
      " '\"patient id: Breast-55\"' '\"patient id: Breast-56\"'\n",
      " '\"patient id: Breast-564\"' '\"patient id: Breast-59\"'\n",
      " '\"patient id: Breast-590\"' '\"patient id: Breast-60\"'\n",
      " '\"patient id: Breast-603\"' '\"patient id: Breast-61\"'\n",
      " '\"patient id: Breast-85\"' '\"patient id: Breast-86\"'\n",
      " '\"patient id: Breast-97\"' '\"patient id: Breast-ALK-82\"'\n",
      " '\"patient id: Chol-292\"' '\"patient id: Chol-316\"'\n",
      " '\"patient id: Chol-341\"' '\"patient id: Chol-376\"'\n",
      " '\"patient id: Chol-379\"' '\"patient id: Chol-410\"'\n",
      " '\"patient id: Chol-442\"' '\"patient id: Chol-460\"'\n",
      " '\"patient id: Chol-611\"' '\"patient id: Chol-ALK-28\"'\n",
      " '\"patient id: CRC-256\"' '\"patient id: CRC-258\"' '\"patient id: CRC-272\"'\n",
      " '\"patient id: CRC-276\"' '\"patient id: CRC-304\"' '\"patient id: CRC-329\"'\n",
      " '\"patient id: CRC-342\"' '\"patient id: CRC-351\"' '\"patient id: CRC-354\"'\n",
      " '\"patient id: CRC-356\"' '\"patient id: CRC-357\"' '\"patient id: CRC-374\"'\n",
      " '\"patient id: CRC-378\"' '\"patient id: CRC-420\"' '\"patient id: CRC-429\"'\n",
      " '\"patient id: CRC-450\"' '\"patient id: CRC-453\"' '\"patient id: CRC-459\"'\n",
      " '\"patient id: CRC-462\"' '\"patient id: CRC-463\"' '\"patient id: CRC-466\"'\n",
      " '\"patient id: CRC-474\"' '\"patient id: CRC-487\"' '\"patient id: CRC-496\"'\n",
      " '\"patient id: CRC-497\"' '\"patient id: CRC-504\"' '\"patient id: CRC-511\"'\n",
      " '\"patient id: CRC-523\"' '\"patient id: CRC-524\"' '\"patient id: CRC-527\"'\n",
      " '\"patient id: CRC-531\"' '\"patient id: CRC-543\"' '\"patient id: CRC-546\"'\n",
      " '\"patient id: CRC-548\"' '\"patient id: CRC-552\"' '\"patient id: CRC-556\"'\n",
      " '\"patient id: CRC-565\"' '\"patient id: CRC-576\"' '\"patient id: GBM-280\"'\n",
      " '\"patient id: GBM-284\"' '\"patient id: GBM-306\"' '\"patient id: GBM-360\"'\n",
      " '\"patient id: GBM-372\"' '\"patient id: GBM-378\"' '\"patient id: GBM-383\"'\n",
      " '\"patient id: GBM-393\"' '\"patient id: GBM-394\"' '\"patient id: GBM-398\"'\n",
      " '\"patient id: GBM-402\"' '\"patient id: GBM-406\"' '\"patient id: GBM-408\"'\n",
      " '\"patient id: GBM-417\"' '\"patient id: GBM-418\"' '\"patient id: GBM-419\"'\n",
      " '\"patient id: GBM-421\"' '\"patient id: GBM-422\"' '\"patient id: GBM-425\"'\n",
      " '\"patient id: GBM-429\"' '\"patient id: GBM-430\"' '\"patient id: GBM-431\"'\n",
      " '\"patient id: GBM-436\"' '\"patient id: GBM-438\"' '\"patient id: GBM-439\"'\n",
      " '\"patient id: GBM-440\"' '\"patient id: GBM-443\"' '\"patient id: GBM-445\"'\n",
      " '\"patient id: GBM-454\"' '\"patient id: GBM-456\"' '\"patient id: GBM-468\"'\n",
      " '\"patient id: GBM-471\"' '\"patient id: GBM-475\"' '\"patient id: GBM-476\"'\n",
      " '\"patient id: GBM-480\"' '\"patient id: GBM-484\"' '\"patient id: GBM-485\"'\n",
      " '\"patient id: GBM-486\"' '\"patient id: GBM-499\"' '\"patient id: GBM-505\"'\n",
      " '\"patient id: HC-01\"' '\"patient id: HC-02\"' '\"patient id: HC-03\"'\n",
      " '\"patient id: HC-04\"' '\"patient id: HC-05\"' '\"patient id: HC-06\"'\n",
      " '\"patient id: HC-07\"' '\"patient id: HC-08\"' '\"patient id: HC-09\"'\n",
      " '\"patient id: HC-10\"' '\"patient id: HC-11\"' '\"patient id: HC-12\"'\n",
      " '\"patient id: HC-13\"' '\"patient id: HC-14\"' '\"patient id: HC-15\"'\n",
      " '\"patient id: HC-16\"' '\"patient id: HC-17\"' '\"patient id: HC-18\"'\n",
      " '\"patient id: HC-19\"' '\"patient id: HC-20\"' '\"patient id: HC-21\"'\n",
      " '\"patient id: HC-22\"' '\"patient id: HC-23\"' '\"patient id: HC-24\"'\n",
      " '\"patient id: HC-25\"' '\"patient id: HC-26\"' '\"patient id: HC-27\"'\n",
      " '\"patient id: HC-28\"' '\"patient id: HC-29\"' '\"patient id: HC-30\"'\n",
      " '\"patient id: HC-31\"' '\"patient id: HC-32\"' '\"patient id: HC-33\"'\n",
      " '\"patient id: HC-34\"' '\"patient id: HC-35\"' '\"patient id: HC-36\"'\n",
      " '\"patient id: HC-37\"' '\"patient id: HC-38\"' '\"patient id: HC-39\"'\n",
      " '\"patient id: HC-40\"' '\"patient id: HC-41\"' '\"patient id: HC-42\"'\n",
      " '\"patient id: HC-43\"' '\"patient id: HC-44\"' '\"patient id: HC-45\"'\n",
      " '\"patient id: HC-46\"' '\"patient id: HC-47\"' '\"patient id: HC-48\"'\n",
      " '\"patient id: HC-49\"' '\"patient id: HC-50\"' '\"patient id: HC-51\"'\n",
      " '\"patient id: HC-52\"' '\"patient id: HC-53\"' '\"patient id: HC-54\"'\n",
      " '\"patient id: HC-55\"' '\"patient id: Liver-274\"' '\"patient id: Liver-297\"'\n",
      " '\"patient id: Liver-366\"' '\"patient id: Lung-0082\"'\n",
      " '\"patient id: Lung-0096\"' '\"patient id: Lung-01\"'\n",
      " '\"patient id: Lung-0100\"' '\"patient id: Lung-0101\"'\n",
      " '\"patient id: Lung-0102\"' '\"patient id: Lung-0104\"'\n",
      " '\"patient id: Lung-0110\"' '\"patient id: Lung-0112\"'\n",
      " '\"patient id: Lung-0119\"' '\"patient id: Lung-04\"' '\"patient id: Lung-06\"'\n",
      " '\"patient id: Lung-07\"' '\"patient id: Lung-08\"' '\"patient id: Lung-12\"'\n",
      " '\"patient id: Lung-14\"' '\"patient id: Lung-15\"' '\"patient id: Lung-16\"'\n",
      " '\"patient id: Lung-17\"' '\"patient id: Lung-20\"' '\"patient id: Lung-20-1\"'\n",
      " '\"patient id: Lung-24-1\"' '\"patient id: Lung-25\"' '\"patient id: Lung-26\"'\n",
      " '\"patient id: Lung-271\"' '\"patient id: Lung-29\"' '\"patient id: Lung-30\"'\n",
      " '\"patient id: Lung-37\"' '\"patient id: Lung-40\"' '\"patient id: Lung-42\"'\n",
      " '\"patient id: Lung-46\"' '\"patient id: Lung-58\"' '\"patient id: Lung-61\"'\n",
      " '\"patient id: Lung-62\"' '\"patient id: Lung-74\"' '\"patient id: Lung-L.13\"'\n",
      " '\"patient id: Lung-L.15\"' '\"patient id: Lung-L.26\"'\n",
      " '\"patient id: Lung-L.33\"' '\"patient id: Lung-L.4\"'\n",
      " '\"patient id: Panc-260\"' '\"patient id: Panc-286\"'\n",
      " '\"patient id: Panc-314\"' '\"patient id: Panc-327\"'\n",
      " '\"patient id: Panc-328\"' '\"patient id: Panc-331\"'\n",
      " '\"patient id: Panc-335\"' '\"patient id: Panc-367\"'\n",
      " '\"patient id: Panc-369\"' '\"patient id: Panc-370\"'\n",
      " '\"patient id: Panc-371\"' '\"patient id: Panc-380\"'\n",
      " '\"patient id: Panc-382\"' '\"patient id: Panc-387\"'\n",
      " '\"patient id: Panc-388\"' '\"patient id: Panc-389\"'\n",
      " '\"patient id: Panc-390\"' '\"patient id: Panc-391\"'\n",
      " '\"patient id: Panc-431\"' '\"patient id: Panc-434\"'\n",
      " '\"patient id: Panc-435\"' '\"patient id: Panc-448\"'\n",
      " '\"patient id: Panc-449\"' '\"patient id: Panc-451\"'\n",
      " '\"patient id: Panc-461\"' '\"patient id: Panc-477\"'\n",
      " '\"patient id: Panc-488\"' '\"patient id: Panc-493\"'\n",
      " '\"patient id: Panc-498\"' '\"patient id: Panc-501\"'\n",
      " '\"patient id: Panc-503\"' '\"patient id: Panc-512\"'\n",
      " '\"patient id: Panc-553\"' '\"patient id: Panc-593\"'\n",
      " '\"patient id: Panc-597\"' '\"patient id: Breast-H11\"'\n",
      " '\"patient id: Breast-H31\"' '\"patient id: Breast-H38\"'\n",
      " '\"patient id: Breast-H59\"' '\"patient id: Breast-H63\"'\n",
      " '\"patient id: Breast-H66\"' '\"patient id: Breast-H68\"'\n",
      " '\"patient id: Breast-H74\"' '\"patient id: Breast-H75\"'\n",
      " '\"patient id: Breast-H76\"' '\"patient id: Breast-H81\"'\n",
      " '\"patient id: Breast-H83\"' '\"patient id: Breast-H84\"'\n",
      " '\"patient id: Breast-H89\"' '\"patient id: Breast-H92\"'\n",
      " '\"patient id: Chol-BRAF4\"' '\"patient id: CRC-368\"'\n",
      " '\"patient id: CRC-412\"' '\"patient id: CRC-BRAF5\"'\n",
      " '\"patient id: CRC-BRAF6\"' '\"patient id: Lung-49\"' '\"patient id: Lung-57\"'\n",
      " '\"patient id: Lung-59\"' '\"patient id: Lung-L01\"' '\"patient id: Lung-L03\"'\n",
      " '\"patient id: Lung-L07\"' '\"patient id: Lung-L11\"'\n",
      " '\"patient id: Lung-L12\"' '\"patient id: Lung-L18\"'\n",
      " '\"patient id: Lung-L19\"' '\"patient id: Lung-L20\"'\n",
      " '\"patient id: Lung-L22\"' '\"patient id: Lung-L23\"'\n",
      " '\"patient id: Lung-L25\"' '\"patient id: Lung-L39\"'\n",
      " '\"patient id: Lung-L40\"' '\"patient id: Lung-L51\"'\n",
      " '\"patient id: Lung-L58\"' '\"patient id: Lung-L59\"'\n",
      " '\"patient id: Lung-L65\"']\n",
      "['\"cancer type: Breast\"' '\"cancer type: Hepatobiliary\"'\n",
      " '\"cancer type: CRC\"' '\"cancer type: GBM\"' '\"cancer type: HC\"'\n",
      " '\"cancer type: Lung\"' '\"cancer type: Pancreas\"']\n",
      "['\"batch: Batch03\"' '\"batch: Batch04\"' '\"batch: Batch02\"'\n",
      " '\"batch: Batch01\"' '\"batch: Batch05\"' '\"batch: Batch06\"']\n",
      "['\"mutational subclass: HER2+\"' '\"mutational subclass: wt\"'\n",
      " '\"mutational subclass: Triple Negative\"'\n",
      " '\"mutational subclass: HER2+, PIK3CA\"' '\"mutational subclass: PIK3CA\"'\n",
      " '\"mutational subclass: KRAS\"' '\"mutational subclass: MET\"'\n",
      " '\"mutational subclass: EGFR, MET\"' '\"mutational subclass: EGFR\"'\n",
      " '\"mutational subclass: KRAS, MET\"']\n"
     ]
    }
   ],
   "source": [
    "unique_tissue = series_df['!Sample_characteristics_ch1'].unique()\n",
    "print(unique_tissue)\n",
    "\n",
    "unique_patient_id = series_df['!Sample_characteristics_ch1.2'].unique()\n",
    "print(unique_patient_id)\n",
    "\n",
    "unique_cancer_types = series_df['!Sample_characteristics_ch1.3'].unique()\n",
    "print(unique_cancer_types)\n",
    "\n",
    "unique_batch_id = series_df['!Sample_characteristics_ch1.4'].unique()\n",
    "print(unique_batch_id)\n",
    "\n",
    "unique_mutation_subclass = series_df['!Sample_characteristics_ch1.5'].unique()\n",
    "print(unique_mutation_subclass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f880a163-3613-403d-96a9-e7405f2e0b55",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "!Sample_geo_accession\n",
      "!Sample_status\n",
      "!Sample_submission_date\n",
      "!Sample_last_update_date\n",
      "!Sample_type\n",
      "!Sample_channel_count\n",
      "!Sample_source_name_ch1\n",
      "!Sample_organism_ch1\n",
      "!Sample_characteristics_ch1\n",
      "!Sample_characteristics_ch1.1\n",
      "!Sample_characteristics_ch1.2\n",
      "!Sample_characteristics_ch1.3\n",
      "!Sample_characteristics_ch1.4\n",
      "!Sample_characteristics_ch1.5\n",
      "!Sample_molecule_ch1\n",
      "!Sample_extract_protocol_ch1\n",
      "!Sample_extract_protocol_ch1.1\n",
      "!Sample_taxid_ch1\n",
      "!Sample_description\n",
      "!Sample_data_processing\n",
      "!Sample_data_processing.1\n",
      "!Sample_data_processing.2\n",
      "!Sample_data_processing.3\n",
      "!Sample_data_processing.4\n",
      "!Sample_data_processing.5\n",
      "!Sample_platform_id\n",
      "!Sample_contact_name\n",
      "!Sample_contact_email\n",
      "!Sample_contact_laboratory\n",
      "!Sample_contact_department\n",
      "!Sample_contact_institute\n",
      "!Sample_contact_address\n",
      "!Sample_contact_city\n",
      "!Sample_contact_zip/postal_code\n",
      "!Sample_contact_country\n",
      "!Sample_data_row_count\n",
      "!Sample_instrument_model\n",
      "!Sample_library_selection\n",
      "!Sample_library_source\n",
      "!Sample_library_strategy\n",
      "!Sample_relation\n",
      "!Sample_relation.1\n",
      "!Sample_supplementary_file_1\n",
      "!series_matrix_table_begin\n",
      "\"ID_REF\"\n",
      "!series_matrix_table_end\n"
     ]
    }
   ],
   "source": [
    "series_col_list = series_df.columns.to_list()\n",
    "\n",
    "for i in series_col_list:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1d55166b-446b-4341-935c-3bbba6d6faf6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_geo_accession</th>\n",
       "      <th>!Sample_status</th>\n",
       "      <th>!Sample_submission_date</th>\n",
       "      <th>!Sample_last_update_date</th>\n",
       "      <th>!Sample_type</th>\n",
       "      <th>!Sample_channel_count</th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_organism_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "      <th>!Sample_molecule_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1</th>\n",
       "      <th>!Sample_extract_protocol_ch1.1</th>\n",
       "      <th>!Sample_taxid_ch1</th>\n",
       "      <th>!Sample_description</th>\n",
       "      <th>!Sample_data_processing</th>\n",
       "      <th>!Sample_data_processing.1</th>\n",
       "      <th>!Sample_data_processing.2</th>\n",
       "      <th>!Sample_data_processing.3</th>\n",
       "      <th>!Sample_data_processing.4</th>\n",
       "      <th>!Sample_data_processing.5</th>\n",
       "      <th>!Sample_platform_id</th>\n",
       "      <th>!Sample_contact_name</th>\n",
       "      <th>!Sample_contact_email</th>\n",
       "      <th>!Sample_contact_laboratory</th>\n",
       "      <th>!Sample_contact_department</th>\n",
       "      <th>!Sample_contact_institute</th>\n",
       "      <th>!Sample_contact_address</th>\n",
       "      <th>!Sample_contact_city</th>\n",
       "      <th>!Sample_contact_zip/postal_code</th>\n",
       "      <th>!Sample_contact_country</th>\n",
       "      <th>!Sample_data_row_count</th>\n",
       "      <th>!Sample_instrument_model</th>\n",
       "      <th>!Sample_library_selection</th>\n",
       "      <th>!Sample_library_source</th>\n",
       "      <th>!Sample_library_strategy</th>\n",
       "      <th>!Sample_relation</th>\n",
       "      <th>!Sample_relation.1</th>\n",
       "      <th>!Sample_supplementary_file_1</th>\n",
       "      <th>!series_matrix_table_begin</th>\n",
       "      <th>\"ID_REF\"</th>\n",
       "      <th>!series_matrix_table_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"GSM1662534\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"3-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-03\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662534\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"GSM1662535\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"8-Breast-WT\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-08\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662535\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"GSM1662536\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"10-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-10\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662536\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"GSM1662537\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"Breast-100\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-100\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch04\"</td>\n",
       "      <td>\"mutational subclass: Triple Negative\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662537\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"GSM1662538\"</td>\n",
       "      <td>\"Public on Oct 30 2015\"</td>\n",
       "      <td>\"Apr 21 2015\"</td>\n",
       "      <td>\"May 15 2019\"</td>\n",
       "      <td>\"SRA\"</td>\n",
       "      <td>\"1\"</td>\n",
       "      <td>\"15-Breast-Her2-ampl\"</td>\n",
       "      <td>\"Homo sapiens\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-15\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "      <td>\"total RNA\"</td>\n",
       "      <td>\"Blood platelets were isolated from whole bloo...</td>\n",
       "      <td>\"100-500 pg of platelet total RNA (Bioanalyzer...</td>\n",
       "      <td>\"9606\"</td>\n",
       "      <td>\"\"</td>\n",
       "      <td>\"Single-end 100 bp reads were subjected to pre...</td>\n",
       "      <td>\"Cleaned-up reads were mapped to the hg19 refe...</td>\n",
       "      <td>\"Selection of intron-spanning reads was perfor...</td>\n",
       "      <td>\"Read summarization was performed using HTSeq ...</td>\n",
       "      <td>\"Genome_build: hg19\"</td>\n",
       "      <td>\"Supplementary_files_format_and_content: Sampl...</td>\n",
       "      <td>\"GPL16791\"</td>\n",
       "      <td>\"Myron,,Best\"</td>\n",
       "      <td>\"m.best@vumc.nl\"</td>\n",
       "      <td>\"Neuro-oncology Research Group\"</td>\n",
       "      <td>\"Neurosurgery\"</td>\n",
       "      <td>\"VU University Medical Center\"</td>\n",
       "      <td>\"De Boelelaan 1117\"</td>\n",
       "      <td>\"Amsterdam\"</td>\n",
       "      <td>\"1081 HV\"</td>\n",
       "      <td>\"Netherlands\"</td>\n",
       "      <td>\"0\"</td>\n",
       "      <td>\"Illumina HiSeq 2500\"</td>\n",
       "      <td>\"cDNA\"</td>\n",
       "      <td>\"transcriptomic\"</td>\n",
       "      <td>\"RNA-Seq\"</td>\n",
       "      <td>\"BioSample: https://www.ncbi.nlm.nih.gov/biosa...</td>\n",
       "      <td>\"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...</td>\n",
       "      <td>\"NONE\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>\"GSM1662538\"</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  !Sample_geo_accession           !Sample_status !Sample_submission_date  \\\n",
       "0          \"GSM1662534\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "1          \"GSM1662535\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "2          \"GSM1662536\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "3          \"GSM1662537\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "4          \"GSM1662538\"  \"Public on Oct 30 2015\"           \"Apr 21 2015\"   \n",
       "\n",
       "  !Sample_last_update_date !Sample_type !Sample_channel_count  \\\n",
       "0            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "1            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "2            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "3            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "4            \"May 15 2019\"        \"SRA\"                   \"1\"   \n",
       "\n",
       "  !Sample_source_name_ch1 !Sample_organism_ch1 !Sample_characteristics_ch1  \\\n",
       "0    \"3-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "1           \"8-Breast-WT\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "2   \"10-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "3            \"Breast-100\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "4   \"15-Breast-Her2-ampl\"       \"Homo sapiens\"             \"tissue: blood\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.1 !Sample_characteristics_ch1.2  \\\n",
       "0     \"cell type: Thrombocytes\"       \"patient id: Breast-03\"   \n",
       "1     \"cell type: Thrombocytes\"       \"patient id: Breast-08\"   \n",
       "2     \"cell type: Thrombocytes\"       \"patient id: Breast-10\"   \n",
       "3     \"cell type: Thrombocytes\"      \"patient id: Breast-100\"   \n",
       "4     \"cell type: Thrombocytes\"       \"patient id: Breast-15\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.3 !Sample_characteristics_ch1.4  \\\n",
       "0         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "1         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "2         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "3         \"cancer type: Breast\"              \"batch: Batch04\"   \n",
       "4         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "\n",
       "            !Sample_characteristics_ch1.5 !Sample_molecule_ch1  \\\n",
       "0            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "1               \"mutational subclass: wt\"          \"total RNA\"   \n",
       "2            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "3  \"mutational subclass: Triple Negative\"          \"total RNA\"   \n",
       "4            \"mutational subclass: HER2+\"          \"total RNA\"   \n",
       "\n",
       "                        !Sample_extract_protocol_ch1  \\\n",
       "0  \"Blood platelets were isolated from whole bloo...   \n",
       "1  \"Blood platelets were isolated from whole bloo...   \n",
       "2  \"Blood platelets were isolated from whole bloo...   \n",
       "3  \"Blood platelets were isolated from whole bloo...   \n",
       "4  \"Blood platelets were isolated from whole bloo...   \n",
       "\n",
       "                      !Sample_extract_protocol_ch1.1 !Sample_taxid_ch1  \\\n",
       "0  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "1  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "2  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "3  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "4  \"100-500 pg of platelet total RNA (Bioanalyzer...            \"9606\"   \n",
       "\n",
       "  !Sample_description                            !Sample_data_processing  \\\n",
       "0                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "1                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "2                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "3                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "4                  \"\"  \"Single-end 100 bp reads were subjected to pre...   \n",
       "\n",
       "                           !Sample_data_processing.1  \\\n",
       "0  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "1  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "2  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "3  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "4  \"Cleaned-up reads were mapped to the hg19 refe...   \n",
       "\n",
       "                           !Sample_data_processing.2  \\\n",
       "0  \"Selection of intron-spanning reads was perfor...   \n",
       "1  \"Selection of intron-spanning reads was perfor...   \n",
       "2  \"Selection of intron-spanning reads was perfor...   \n",
       "3  \"Selection of intron-spanning reads was perfor...   \n",
       "4  \"Selection of intron-spanning reads was perfor...   \n",
       "\n",
       "                           !Sample_data_processing.3  \\\n",
       "0  \"Read summarization was performed using HTSeq ...   \n",
       "1  \"Read summarization was performed using HTSeq ...   \n",
       "2  \"Read summarization was performed using HTSeq ...   \n",
       "3  \"Read summarization was performed using HTSeq ...   \n",
       "4  \"Read summarization was performed using HTSeq ...   \n",
       "\n",
       "  !Sample_data_processing.4  \\\n",
       "0      \"Genome_build: hg19\"   \n",
       "1      \"Genome_build: hg19\"   \n",
       "2      \"Genome_build: hg19\"   \n",
       "3      \"Genome_build: hg19\"   \n",
       "4      \"Genome_build: hg19\"   \n",
       "\n",
       "                           !Sample_data_processing.5 !Sample_platform_id  \\\n",
       "0  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "1  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "2  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "3  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "4  \"Supplementary_files_format_and_content: Sampl...          \"GPL16791\"   \n",
       "\n",
       "  !Sample_contact_name !Sample_contact_email       !Sample_contact_laboratory  \\\n",
       "0        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "1        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "2        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "3        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "4        \"Myron,,Best\"      \"m.best@vumc.nl\"  \"Neuro-oncology Research Group\"   \n",
       "\n",
       "  !Sample_contact_department       !Sample_contact_institute  \\\n",
       "0             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "1             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "2             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "3             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "4             \"Neurosurgery\"  \"VU University Medical Center\"   \n",
       "\n",
       "  !Sample_contact_address !Sample_contact_city  \\\n",
       "0     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "1     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "2     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "3     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "4     \"De Boelelaan 1117\"          \"Amsterdam\"   \n",
       "\n",
       "  !Sample_contact_zip/postal_code !Sample_contact_country  \\\n",
       "0                       \"1081 HV\"           \"Netherlands\"   \n",
       "1                       \"1081 HV\"           \"Netherlands\"   \n",
       "2                       \"1081 HV\"           \"Netherlands\"   \n",
       "3                       \"1081 HV\"           \"Netherlands\"   \n",
       "4                       \"1081 HV\"           \"Netherlands\"   \n",
       "\n",
       "  !Sample_data_row_count !Sample_instrument_model !Sample_library_selection  \\\n",
       "0                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "1                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "2                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "3                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "4                    \"0\"    \"Illumina HiSeq 2500\"                    \"cDNA\"   \n",
       "\n",
       "  !Sample_library_source !Sample_library_strategy  \\\n",
       "0       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "1       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "2       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "3       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "4       \"transcriptomic\"                \"RNA-Seq\"   \n",
       "\n",
       "                                    !Sample_relation  \\\n",
       "0  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "1  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "2  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "3  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "4  \"BioSample: https://www.ncbi.nlm.nih.gov/biosa...   \n",
       "\n",
       "                                  !Sample_relation.1  \\\n",
       "0  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "1  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "2  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "3  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "4  \"SRA: https://www.ncbi.nlm.nih.gov/sra?term=SR...   \n",
       "\n",
       "  !Sample_supplementary_file_1  !series_matrix_table_begin      \"ID_REF\"  \\\n",
       "0                       \"NONE\"                         NaN  \"GSM1662534\"   \n",
       "1                       \"NONE\"                         NaN  \"GSM1662535\"   \n",
       "2                       \"NONE\"                         NaN  \"GSM1662536\"   \n",
       "3                       \"NONE\"                         NaN  \"GSM1662537\"   \n",
       "4                       \"NONE\"                         NaN  \"GSM1662538\"   \n",
       "\n",
       "   !series_matrix_table_end  \n",
       "0                       NaN  \n",
       "1                       NaN  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                       NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "\n",
    "series_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "84e83f9e-d747-45f2-a464-d774056436d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df1 = series_df.drop([\"!series_matrix_table_begin\", \"!series_matrix_table_end\"], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e7abef9d-970a-437b-82c1-ba149dc8446f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 285 entries, 0 to 284\n",
      "Data columns (total 44 columns):\n",
      " #   Column                           Non-Null Count  Dtype \n",
      "---  ------                           --------------  ----- \n",
      " 0   !Sample_geo_accession            285 non-null    object\n",
      " 1   !Sample_status                   285 non-null    object\n",
      " 2   !Sample_submission_date          285 non-null    object\n",
      " 3   !Sample_last_update_date         285 non-null    object\n",
      " 4   !Sample_type                     285 non-null    object\n",
      " 5   !Sample_channel_count            285 non-null    object\n",
      " 6   !Sample_source_name_ch1          285 non-null    object\n",
      " 7   !Sample_organism_ch1             285 non-null    object\n",
      " 8   !Sample_characteristics_ch1      285 non-null    object\n",
      " 9   !Sample_characteristics_ch1.1    285 non-null    object\n",
      " 10  !Sample_characteristics_ch1.2    285 non-null    object\n",
      " 11  !Sample_characteristics_ch1.3    285 non-null    object\n",
      " 12  !Sample_characteristics_ch1.4    285 non-null    object\n",
      " 13  !Sample_characteristics_ch1.5    285 non-null    object\n",
      " 14  !Sample_molecule_ch1             285 non-null    object\n",
      " 15  !Sample_extract_protocol_ch1     285 non-null    object\n",
      " 16  !Sample_extract_protocol_ch1.1   285 non-null    object\n",
      " 17  !Sample_taxid_ch1                285 non-null    object\n",
      " 18  !Sample_description              285 non-null    object\n",
      " 19  !Sample_data_processing          285 non-null    object\n",
      " 20  !Sample_data_processing.1        285 non-null    object\n",
      " 21  !Sample_data_processing.2        285 non-null    object\n",
      " 22  !Sample_data_processing.3        285 non-null    object\n",
      " 23  !Sample_data_processing.4        285 non-null    object\n",
      " 24  !Sample_data_processing.5        285 non-null    object\n",
      " 25  !Sample_platform_id              285 non-null    object\n",
      " 26  !Sample_contact_name             285 non-null    object\n",
      " 27  !Sample_contact_email            285 non-null    object\n",
      " 28  !Sample_contact_laboratory       285 non-null    object\n",
      " 29  !Sample_contact_department       285 non-null    object\n",
      " 30  !Sample_contact_institute        285 non-null    object\n",
      " 31  !Sample_contact_address          285 non-null    object\n",
      " 32  !Sample_contact_city             285 non-null    object\n",
      " 33  !Sample_contact_zip/postal_code  285 non-null    object\n",
      " 34  !Sample_contact_country          285 non-null    object\n",
      " 35  !Sample_data_row_count           285 non-null    object\n",
      " 36  !Sample_instrument_model         285 non-null    object\n",
      " 37  !Sample_library_selection        285 non-null    object\n",
      " 38  !Sample_library_source           285 non-null    object\n",
      " 39  !Sample_library_strategy         285 non-null    object\n",
      " 40  !Sample_relation                 285 non-null    object\n",
      " 41  !Sample_relation.1               285 non-null    object\n",
      " 42  !Sample_supplementary_file_1     285 non-null    object\n",
      " 43  \"ID_REF\"                         285 non-null    object\n",
      "dtypes: object(44)\n",
      "memory usage: 98.1+ KB\n"
     ]
    }
   ],
   "source": [
    "series_df1.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0869b3be-ef31-4bc5-a555-bb4b3c87853f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>!Sample_source_name_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1</th>\n",
       "      <th>!Sample_characteristics_ch1.1</th>\n",
       "      <th>!Sample_characteristics_ch1.2</th>\n",
       "      <th>!Sample_characteristics_ch1.3</th>\n",
       "      <th>!Sample_characteristics_ch1.4</th>\n",
       "      <th>!Sample_characteristics_ch1.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"3-Breast-Her2-ampl\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-03\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"8-Breast-WT\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-08\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: wt\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"10-Breast-Her2-ampl\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-10\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"Breast-100\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-100\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch04\"</td>\n",
       "      <td>\"mutational subclass: Triple Negative\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"15-Breast-Her2-ampl\"</td>\n",
       "      <td>\"tissue: blood\"</td>\n",
       "      <td>\"cell type: Thrombocytes\"</td>\n",
       "      <td>\"patient id: Breast-15\"</td>\n",
       "      <td>\"cancer type: Breast\"</td>\n",
       "      <td>\"batch: Batch03\"</td>\n",
       "      <td>\"mutational subclass: HER2+\"</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  !Sample_source_name_ch1 !Sample_characteristics_ch1  \\\n",
       "0    \"3-Breast-Her2-ampl\"             \"tissue: blood\"   \n",
       "1           \"8-Breast-WT\"             \"tissue: blood\"   \n",
       "2   \"10-Breast-Her2-ampl\"             \"tissue: blood\"   \n",
       "3            \"Breast-100\"             \"tissue: blood\"   \n",
       "4   \"15-Breast-Her2-ampl\"             \"tissue: blood\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.1 !Sample_characteristics_ch1.2  \\\n",
       "0     \"cell type: Thrombocytes\"       \"patient id: Breast-03\"   \n",
       "1     \"cell type: Thrombocytes\"       \"patient id: Breast-08\"   \n",
       "2     \"cell type: Thrombocytes\"       \"patient id: Breast-10\"   \n",
       "3     \"cell type: Thrombocytes\"      \"patient id: Breast-100\"   \n",
       "4     \"cell type: Thrombocytes\"       \"patient id: Breast-15\"   \n",
       "\n",
       "  !Sample_characteristics_ch1.3 !Sample_characteristics_ch1.4  \\\n",
       "0         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "1         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "2         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "3         \"cancer type: Breast\"              \"batch: Batch04\"   \n",
       "4         \"cancer type: Breast\"              \"batch: Batch03\"   \n",
       "\n",
       "            !Sample_characteristics_ch1.5  \n",
       "0            \"mutational subclass: HER2+\"  \n",
       "1               \"mutational subclass: wt\"  \n",
       "2            \"mutational subclass: HER2+\"  \n",
       "3  \"mutational subclass: Triple Negative\"  \n",
       "4            \"mutational subclass: HER2+\"  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series_df2 = series_df[[\"!Sample_source_name_ch1\", \"!Sample_characteristics_ch1\", \"!Sample_characteristics_ch1.1\", \"!Sample_characteristics_ch1.2\", \"!Sample_characteristics_ch1.3\",\n",
    "\"!Sample_characteristics_ch1.4\", \"!Sample_characteristics_ch1.5\"]]\n",
    "\n",
    "series_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0da11089-0cf1-4155-a121-1a22302accab",
   "metadata": {},
   "outputs": [],
   "source": [
    "# series_df2.rename(columns={\"!Sample_characteristics_ch1\": \"tissue\", \"!Sample_characteristics_ch1.1\": \"cell_type\",\n",
    "#                           \"!Sample_characteristics_ch1.2\": \"patient_id\", \"!Sample_characteristics_ch1.3\": \"cancer_type\",\n",
    "#                           \"!Sample_characteristics_ch1.4\": \"batch\", \"!Sample_characteristics_ch1.5\": \"mutational_subclass\",\n",
    "#                           \"!Sample_source_name_ch1\": \"unique_id\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c0b3f4df-447f-4d9a-aa1e-993283dc2a77",
   "metadata": {},
   "outputs": [],
   "source": [
    "# series_df2[\"unique_id\"] = series_df2[\"unique_id\"].astype(str).str.replace('\"', '')\n",
    "# series_df2[\"patient_id\"] = series_df2[\"patient_id\"].astype(str).str.replace('\"', '').str.strip()\n",
    "# series_df2[\"cancer_type\"] = series_df2[\"cancer_type\"].astype(str).str.replace('\"', '').str.strip()\n",
    "# series_df2[\"mutational_subclass\"] = series_df2[\"mutational_subclass\"].astype(str).str.replace('\"', '').str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "32f51962-123e-4938-a141-c1a701afcd81",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df2 = series_df2.rename(columns={\n",
    "    \"!Sample_characteristics_ch1\": \"tissue\",\n",
    "    \"!Sample_characteristics_ch1.1\": \"cell_type\",\n",
    "    \"!Sample_characteristics_ch1.2\": \"patient_id\",\n",
    "    \"!Sample_characteristics_ch1.3\": \"cancer_type\",\n",
    "    \"!Sample_characteristics_ch1.4\": \"batch\",\n",
    "    \"!Sample_characteristics_ch1.5\": \"mutational_subclass\",\n",
    "    \"!Sample_source_name_ch1\": \"unique_id\"\n",
    "})\n",
    "\n",
    "series_df2.to_csv(\"series_df2_renamed.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f5688be6-25bb-4538-a307-542bd8737253",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df2[\"unique_id\"] = series_df2[\"unique_id\"].astype(str).str.replace('\"', '')\n",
    "series_df2[\"patient_id\"] = series_df2[\"patient_id\"].astype(str).str.replace('\"', '').str.strip()\n",
    "series_df2[\"cancer_type\"] = series_df2[\"cancer_type\"].astype(str).str.replace('\"', '').str.strip()\n",
    "series_df2[\"mutational_subclass\"] = series_df2[\"mutational_subclass\"].astype(str).str.replace('\"', '').str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9a79a11f-1ce1-412d-abc9-b58fb99d60a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "series_df2.to_csv(\"Preprossed_GSE68086_series_matrix_fixed.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5a9d9460-0b65-4b67-82be-959fbe117604",
   "metadata": {},
   "outputs": [],
   "source": [
    "#########################   TEMP_DF  #######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "ba35256c-be8b-4b8d-bb15-8c2ca377efcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "tep_df = pd.read_csv(\"GSE68086_TEP_data_matrix.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "fb1aff6b-0341-4da5-ab3e-ad094fb3266a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 57736 entries, 0 to 57735\n",
      "Columns: 286 entries, Unnamed: 0 to MGH-NSCLC-L65-TR523\n",
      "dtypes: int64(285), object(1)\n",
      "memory usage: 126.0+ MB\n"
     ]
    }
   ],
   "source": [
    "tep_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "24bc0081-8238-4a94-ade9-c050d44171ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "#####transposed tep_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "5632a7ec-7d3e-484b-8b1c-004c8cbd01ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "tep_transposed = tep_df.transpose()\n",
    "tep_transposed.to_csv('tep_transposed_data.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "111a1f99-7c95-4c8e-a58c-8f1f066023ea",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b345cf9b-653e-4729-8213-b2def5422b2a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "457febc3-c1c7-4120-88fb-1f333aacf179",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "018ff865-199e-416f-ac09-1dd1f4908526",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09327eec-f899-4f6f-97b4-d2e482d7e13a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "501c7699-1a23-4e7d-b8c6-2ed66bbc746b",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03bdc58b-0039-4932-bcf1-95501e5cd087",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c79cad72-490e-47c0-9936-a5295ec662d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
